<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Shashank Tiwari" />


<title>Analysis of MeDip-Seq data</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.27/datatables.js"></script>
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chicane-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="coding.html">Noncoding split</a>
</li>
<li>
  <a href="norm.html">RNA length Normalised</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of MeDip-Seq data</h1>
<h4 class="author">Shashank Tiwari</h4>
<h4 class="date">Most recent update:<br> 2023-07-12</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#root.dir &lt;- here::here()</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">collapse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">comment =</span> <span class="st">&quot;#&gt;&quot;</span>,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#root.dir = root.dir</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">fig.height =</span> <span class="dv">6</span>,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">fig.width =</span> <span class="fl">7.00787</span> <span class="co">#178mm</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>)  </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_knit<span class="sc">$</span><span class="fu">set</span>(<span class="co">#root.dir = &quot;~/Radicl_Seq/Radicl_seq_functions/&quot;, </span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dpi =</span> <span class="dv">350</span>)  </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## Warning: package &#39;stringr&#39; was built under R version 4.2.2</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsamtools)</span></code></pre></div>
<pre><code>## Warning: package &#39;BiocGenerics&#39; was built under R version 4.2.1</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span></code></pre></div>
<pre><code>## Warning: package &#39;rtracklayer&#39; was built under R version 4.2.2</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code></pre></div>
<pre><code>## Warning: package &#39;viridisLite&#39; was built under R version 4.2.1</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomation)</span></code></pre></div>
<pre><code>## Warning: replacing previous import &#39;Biostrings::pattern&#39; by &#39;grid::pattern&#39;
## when loading &#39;genomation&#39;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magick)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(slickR)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svglite)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bedtoolsr)</span></code></pre></div>
<pre><code>## Warning in value[[3L]](cond): bedtools does not appear to be installed or not in your PATH. If it is installed, please add it to your PATH or run:
## options(bedtools.path = &quot;/path/to/&quot;)</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#library(ggpubr)</span></span></code></pre></div>
<pre><code>#&gt;             used   (Mb) gc trigger   (Mb)  max used   (Mb)
#&gt; Ncells  10550015  563.5   16736215  893.9  11460993  612.1
#&gt; Vcells 803487402 6130.2 1054181735 8042.8 803550831 6130.7
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;73%&quot;        &quot;27%&quot;        &quot;16%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;67%&quot;        &quot;33%&quot;         &quot;2%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;77%&quot;        &quot;23%&quot;        &quot;17%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;71%&quot;        &quot;29%&quot;         &quot;1%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;72%&quot;        &quot;28%&quot;        &quot;15%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;66%&quot;        &quot;34%&quot;         &quot;2%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;73%&quot;        &quot;27%&quot;         &quot;3%&quot; 
#&gt;        trans          cis bait.to.bait 
#&gt;        &quot;70%&quot;        &quot;30%&quot;         &quot;2%&quot;</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>sktech <span class="ot">=</span> htmltools<span class="sc">::</span><span class="fu">withTags</span>(<span class="fu">table</span>(</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">class =</span> <span class="st">&#39;display&#39;</span>,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">thead</span>(</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tr</span>(</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">th</span>(<span class="at">rowspan =</span> <span class="dv">2</span>, <span class="st">&#39;Type&#39;</span>),</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">th</span>(<span class="at">colspan =</span> <span class="dv">5</span>, <span class="st">&#39;All&#39;</span>),</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">th</span>(<span class="at">colspan =</span> <span class="dv">5</span>, <span class="st">&#39;Significant&#39;</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tr</span>(</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">lapply</span>( <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;Ctrl Coding&quot;</span>, <span class="st">&quot;Ctrl NonCoding&quot;</span>,<span class="st">&quot;ICM Coding&quot;</span>,<span class="st">&quot;ICM NonCoding&quot;</span>),<span class="dv">2</span>), th)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(qc_df, <span class="at">container =</span> sktech, <span class="at">rownames =</span> <span class="cn">TRUE</span>,<span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX=</span><span class="cn">TRUE</span>))</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-3bfba8b67f270d80eec5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3bfba8b67f270d80eec5">{"x":{"filter":"none","vertical":false,"class":"display","data":[["trans","cis","bait.to.bait"],["73%","27%","16%"],["67%","33%","2%"],["77%","23%","17%"],["71%","29%","1%"],["72%","28%","15%"],["66%","34%","2%"],["73%","27%","3%"],["70%","30%","2%"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th rowspan=\"2\">Type<\/th>\n      <th colspan=\"5\">All<\/th>\n      <th colspan=\"5\">Significant<\/th>\n    <\/tr>\n    <tr>\n      <th>Ctrl Coding<\/th>\n      <th>Ctrl NonCoding<\/th>\n      <th>ICM Coding<\/th>\n      <th>ICM NonCoding<\/th>\n      <th>Ctrl Coding<\/th>\n      <th>Ctrl NonCoding<\/th>\n      <th>ICM Coding<\/th>\n      <th>ICM NonCoding<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="preprocessing" class="section level2">
<h2>Preprocessing</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>control_coding<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="st">&quot;Control Coding&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>control_nc<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="st">&quot;Control NonCoding&quot;</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>ICM_coding<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="st">&quot;ICM Coding&quot;</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>ICM_nc<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="st">&quot;ICM NonCoding&quot;</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>merged_runs <span class="ot">&lt;-</span> <span class="fu">rbind</span>(control_coding, control_nc,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                     ICM_coding,ICM_nc)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>merged_runs<span class="sc">$</span>signif <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(merged_runs<span class="sc">$</span>q.value<span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="st">&quot;significant&quot;</span>,<span class="st">&quot;non-significant&quot;</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>merged_runs<span class="sc">$</span>frag.length <span class="ot">&lt;-</span> merged_runs<span class="sc">$</span>target.end<span class="sc">-</span>merged_runs<span class="sc">$</span>target.start</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>merged_runs<span class="sc">$</span>bait.length <span class="ot">&lt;-</span> merged_runs<span class="sc">$</span>bait.end<span class="sc">-</span>merged_runs<span class="sc">$</span>bait.start</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co">#make bins split res by qvalue bins</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>merged_runs <span class="ot">&lt;-</span> merged_runs[<span class="sc">!</span><span class="fu">is.na</span>(merged_runs<span class="sc">$</span>q.value),]</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>merged_runs[,q_bin<span class="sc">:</span><span class="er">=</span><span class="fu">cut</span>(q.value, <span class="at">breaks=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.05</span>,<span class="fl">0.2</span>,<span class="fl">0.4</span>,<span class="fl">0.6</span>,<span class="fl">0.8</span>,<span class="dv">1</span>),<span class="at">include.lowest=</span><span class="cn">TRUE</span>),]</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>merged_stretch_cis <span class="ot">&lt;-</span> merged_runs[<span class="sc">!</span><span class="fu">is.na</span>(distance),.(<span class="at">count2=</span><span class="dv">1</span><span class="sc">:</span>count),<span class="fu">names</span>(merged_runs)] </span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="co">#get gene names</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>baits_coding <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;~/Radicl_Seq/coding_genes_no_ovrlps.bed&quot;</span>)</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(baits_coding) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>,<span class="st">&quot;bait_name&quot;</span>)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>baits_coding<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;^&quot;</span>,<span class="st">&quot;chr&quot;</span>, baits_coding<span class="sc">$</span>chr)</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>baits_noncoding <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;~/Radicl_Seq/non_coding_genes_no_ovrlps.bed&quot;</span>)</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(baits_noncoding) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>,<span class="st">&quot;bait_name&quot;</span>)</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>baits_noncoding<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;^&quot;</span>,<span class="st">&quot;chr&quot;</span>,baits_noncoding<span class="sc">$</span>chr)</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>baits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(baits_coding, baits_noncoding)</span></code></pre></div>
<pre><code>#&gt;              used   (Mb) gc trigger    (Mb)   max used    (Mb)
#&gt; Ncells   10642417  568.4   16736215   893.9   16736215   893.9
#&gt; Vcells 1195628266 9122.0 3232766879 24664.1 2881604059 21984.9</code></pre>
</div>
<div id="relationship-between-rna-length-and-detected-interactions"
class="section level2">
<h2>Relationship between RNA length and detected interactions</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>merged_runs[,bait_name <span class="sc">:</span><span class="er">=</span> baits<span class="sc">$</span>bait_name[<span class="fu">match</span>(merged_runs<span class="sc">$</span>bait.start,baits<span class="sc">$</span>start)]]</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>merged_runs_non_b2b <span class="ot">&lt;-</span> merged_runs[bait.to.bait<span class="sc">==</span><span class="cn">FALSE</span>]</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>merged_runs_b2b <span class="ot">&lt;-</span> merged_runs[bait.to.bait<span class="sc">==</span><span class="cn">TRUE</span>]</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setorder</span>(merged_runs_non_b2b,q.value)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#make correlation test</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b<span class="sc">$</span>bait.length,merged_runs_non_b2b<span class="sc">$</span>q.value,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>, <span class="at">exact =</span> F)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b$bait.length and merged_runs_non_b2b$q.value</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.05e+22, p-value &lt; 2.2e-16</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        rho </span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.01658304</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b<span class="sc">$</span>bait.length, merged_runs_non_b2b<span class="sc">$</span>count,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>, <span class="at">exact =</span> F)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b$bait.length and merged_runs_non_b2b$count</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0278e+22, p-value &lt; 2.2e-16</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        rho </span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.02724338</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(merged_runs_non_b2b[frag.length<span class="sc">==</span><span class="dv">1999</span>])<span class="sc">/</span><span class="fu">nrow</span>(merged_runs_non_b2b)</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 3.701272e-05</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b<span class="sc">$</span>frag.length, merged_runs_non_b2b<span class="sc">$</span>count,</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact =</span> F)</span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b$frag.length and merged_runs_non_b2b$count</span></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0889e+22, p-value &lt; 2.2e-16</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          rho </span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -0.002075754</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b<span class="sc">$</span>frag.length, merged_runs_non_b2b<span class="sc">$</span>q.value,</span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b$frag.length and merged_runs_non_b2b$q.value</span></span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0836e+22, p-value = 0.0006038</span></span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          rho </span></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.0004850166</span></span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b[frag.length<span class="sc">!=</span><span class="dv">1999</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>         merged_runs_non_b2b[frag.length<span class="sc">!=</span><span class="dv">1999</span>]<span class="sc">$</span>count,</span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-68"><a href="#cb18-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[frag.length != 1999]$frag.length and merged_runs_non_b2b[frag.length != 1999]$count</span></span>
<span id="cb18-69"><a href="#cb18-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0884e+22, p-value &lt; 2.2e-16</span></span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          rho </span></span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -0.001969395</span></span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b[frag.length<span class="sc">!=</span><span class="dv">1999</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a>         merged_runs_non_b2b[frag.length<span class="sc">!=</span><span class="dv">1999</span>]<span class="sc">$</span>q.value,</span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-80"><a href="#cb18-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-81"><a href="#cb18-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-82"><a href="#cb18-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[frag.length != 1999]$frag.length and merged_runs_non_b2b[frag.length != 1999]$q.value</span></span>
<span id="cb18-83"><a href="#cb18-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0833e+22, p-value = 0.0004415</span></span>
<span id="cb18-84"><a href="#cb18-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-85"><a href="#cb18-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-86"><a href="#cb18-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          rho </span></span>
<span id="cb18-87"><a href="#cb18-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.0004969066</span></span>
<span id="cb18-88"><a href="#cb18-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-89"><a href="#cb18-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-90"><a href="#cb18-90" aria-hidden="true" tabindex="-1"></a><span class="co">#check bait to bait</span></span>
<span id="cb18-91"><a href="#cb18-91" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_b2b<span class="sc">$</span>frag.length, merged_runs_b2b<span class="sc">$</span>q.value,</span>
<span id="cb18-92"><a href="#cb18-92" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-93"><a href="#cb18-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-94"><a href="#cb18-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-95"><a href="#cb18-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-96"><a href="#cb18-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_b2b$frag.length and merged_runs_b2b$q.value</span></span>
<span id="cb18-97"><a href="#cb18-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 1.7764e+19, p-value &lt; 2.2e-16</span></span>
<span id="cb18-98"><a href="#cb18-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-99"><a href="#cb18-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-100"><a href="#cb18-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         rho </span></span>
<span id="cb18-101"><a href="#cb18-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.009396152</span></span>
<span id="cb18-102"><a href="#cb18-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-103"><a href="#cb18-103" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_b2b<span class="sc">$</span>frag.length, merged_runs_b2b<span class="sc">$</span>count,</span>
<span id="cb18-104"><a href="#cb18-104" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-105"><a href="#cb18-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-106"><a href="#cb18-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-107"><a href="#cb18-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-108"><a href="#cb18-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_b2b$frag.length and merged_runs_b2b$count</span></span>
<span id="cb18-109"><a href="#cb18-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 1.7734e+19, p-value &lt; 2.2e-16</span></span>
<span id="cb18-110"><a href="#cb18-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-111"><a href="#cb18-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-112"><a href="#cb18-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        rho </span></span>
<span id="cb18-113"><a href="#cb18-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.01108874</span></span>
<span id="cb18-114"><a href="#cb18-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-115"><a href="#cb18-115" aria-hidden="true" tabindex="-1"></a><span class="co">#check non-b2b without cis</span></span>
<span id="cb18-116"><a href="#cb18-116" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b[target.chr<span class="sc">==</span>bait.chr]<span class="sc">$</span>bait.length,</span>
<span id="cb18-117"><a href="#cb18-117" aria-hidden="true" tabindex="-1"></a>         merged_runs_non_b2b[target.chr<span class="sc">==</span>bait.chr]<span class="sc">$</span>q.value,</span>
<span id="cb18-118"><a href="#cb18-118" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-119"><a href="#cb18-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-120"><a href="#cb18-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-121"><a href="#cb18-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-122"><a href="#cb18-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[target.chr == bait.chr]$bait.length and merged_runs_non_b2b[target.chr == bait.chr]$q.value</span></span>
<span id="cb18-123"><a href="#cb18-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 4.3059e+20, p-value &lt; 2.2e-16</span></span>
<span id="cb18-124"><a href="#cb18-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-125"><a href="#cb18-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-126"><a href="#cb18-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        rho </span></span>
<span id="cb18-127"><a href="#cb18-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.01492415</span></span>
<span id="cb18-128"><a href="#cb18-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-129"><a href="#cb18-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-130"><a href="#cb18-130" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b[target.chr<span class="sc">==</span>bait.chr]<span class="sc">$</span>bait.length,</span>
<span id="cb18-131"><a href="#cb18-131" aria-hidden="true" tabindex="-1"></a>         merged_runs_non_b2b[target.chr<span class="sc">==</span>bait.chr]<span class="sc">$</span>count,</span>
<span id="cb18-132"><a href="#cb18-132" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-133"><a href="#cb18-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-134"><a href="#cb18-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-135"><a href="#cb18-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-136"><a href="#cb18-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[target.chr == bait.chr]$bait.length and merged_runs_non_b2b[target.chr == bait.chr]$count</span></span>
<span id="cb18-137"><a href="#cb18-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 4.2591e+20, p-value &lt; 2.2e-16</span></span>
<span id="cb18-138"><a href="#cb18-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-139"><a href="#cb18-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-140"><a href="#cb18-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       rho </span></span>
<span id="cb18-141"><a href="#cb18-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.0256511</span></span>
<span id="cb18-142"><a href="#cb18-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-143"><a href="#cb18-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-144"><a href="#cb18-144" aria-hidden="true" tabindex="-1"></a><span class="co">#raw pvalue</span></span>
<span id="cb18-145"><a href="#cb18-145" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b[frag.length<span class="sc">!=</span><span class="dv">1999</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb18-146"><a href="#cb18-146" aria-hidden="true" tabindex="-1"></a>         merged_runs_non_b2b[frag.length<span class="sc">!=</span><span class="dv">1999</span>]<span class="sc">$</span>p.value,</span>
<span id="cb18-147"><a href="#cb18-147" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-148"><a href="#cb18-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-149"><a href="#cb18-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-150"><a href="#cb18-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-151"><a href="#cb18-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[frag.length != 1999]$frag.length and merged_runs_non_b2b[frag.length != 1999]$p.value</span></span>
<span id="cb18-152"><a href="#cb18-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0837e+22, p-value = 0.02153</span></span>
<span id="cb18-153"><a href="#cb18-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-154"><a href="#cb18-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-155"><a href="#cb18-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          rho </span></span>
<span id="cb18-156"><a href="#cb18-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.0003250403</span></span>
<span id="cb18-157"><a href="#cb18-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-158"><a href="#cb18-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-159"><a href="#cb18-159" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(merged_runs_non_b2b<span class="sc">$</span>frag.length,merged_runs_non_b2b<span class="sc">$</span>p.value,</span>
<span id="cb18-160"><a href="#cb18-160" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-161"><a href="#cb18-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-162"><a href="#cb18-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-163"><a href="#cb18-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-164"><a href="#cb18-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b$frag.length and merged_runs_non_b2b$p.value</span></span>
<span id="cb18-165"><a href="#cb18-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 2.0839e+22, p-value = 0.03068</span></span>
<span id="cb18-166"><a href="#cb18-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-167"><a href="#cb18-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-168"><a href="#cb18-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          rho </span></span>
<span id="cb18-169"><a href="#cb18-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.0003056158</span></span>
<span id="cb18-170"><a href="#cb18-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-171"><a href="#cb18-171" aria-hidden="true" tabindex="-1"></a><span class="co">#summarised counts per fragment</span></span>
<span id="cb18-172"><a href="#cb18-172" aria-hidden="true" tabindex="-1"></a>bait_count <span class="ot">&lt;-</span> merged_runs_non_b2b[,<span class="fu">sum</span>(count),by<span class="ot">=</span>.(bait.id,bait.length)]</span>
<span id="cb18-173"><a href="#cb18-173" aria-hidden="true" tabindex="-1"></a>merged_runs_sum_non_b2b <span class="ot">&lt;-</span> </span>
<span id="cb18-174"><a href="#cb18-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor.test</span>(bait_count<span class="sc">$</span>bait.length,bait_count<span class="sc">$</span>V1,</span>
<span id="cb18-175"><a href="#cb18-175" aria-hidden="true" tabindex="-1"></a>           <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-176"><a href="#cb18-176" aria-hidden="true" tabindex="-1"></a>frag_count <span class="ot">&lt;-</span> merged_runs_non_b2b[,<span class="fu">sum</span>(count),by<span class="ot">=</span>.(target.id,frag.length)]</span>
<span id="cb18-177"><a href="#cb18-177" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(frag_count<span class="sc">$</span>frag.length,frag_count<span class="sc">$</span>V1,</span>
<span id="cb18-178"><a href="#cb18-178" aria-hidden="true" tabindex="-1"></a>         <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span>
<span id="cb18-179"><a href="#cb18-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-180"><a href="#cb18-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb18-181"><a href="#cb18-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-182"><a href="#cb18-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  frag_count$frag.length and frag_count$V1</span></span>
<span id="cb18-183"><a href="#cb18-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 1.7632e+17, p-value &lt; 2.2e-16</span></span>
<span id="cb18-184"><a href="#cb18-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb18-185"><a href="#cb18-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb18-186"><a href="#cb18-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       rho </span></span>
<span id="cb18-187"><a href="#cb18-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.0966443</span></span>
<span id="cb18-188"><a href="#cb18-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-189"><a href="#cb18-189" aria-hidden="true" tabindex="-1"></a><span class="co">#bait to bait interactions</span></span>
<span id="cb18-190"><a href="#cb18-190" aria-hidden="true" tabindex="-1"></a>bait_count2 <span class="ot">&lt;-</span> merged_runs_b2b[,<span class="fu">sum</span>(count),by<span class="ot">=</span>.(bait.id,bait.length)]</span>
<span id="cb18-191"><a href="#cb18-191" aria-hidden="true" tabindex="-1"></a>merged_run_sum_frag_b2b <span class="ot">&lt;-</span> </span>
<span id="cb18-192"><a href="#cb18-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor.test</span>(bait_count2<span class="sc">$</span>bait.length,bait_count2<span class="sc">$</span>V1,</span>
<span id="cb18-193"><a href="#cb18-193" aria-hidden="true" tabindex="-1"></a>           <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>,<span class="at">exact=</span>F)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Print summarized counts by bait where it wasnt bait to bait</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(merged_runs_sum_non_b2b)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  bait_count$bait.length and bait_count$V1</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 4.3434e+12, p-value &lt; 2.2e-16</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       rho </span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.6670276</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Print summ counts by bait where it is bait to bait</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(merged_run_sum_frag_b2b)</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Spearman&#39;s rank correlation rho</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  bait_count2$bait.length and bait_count2$V1</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; S = 1.6347e+12, p-value &lt; 2.2e-16</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       rho </span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.6688893</span></span></code></pre></div>
<p>Trying wilcoxons test</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#non bait to bait</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>non_b2b_bait_wt <span class="ot">&lt;-</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wilcox.test</span>(merged_runs_non_b2b[q.value<span class="sc">&lt;</span><span class="fl">0.05</span>]<span class="sc">$</span>bait.length,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>              merged_runs_non_b2b[q.value<span class="sc">&gt;=</span><span class="fl">0.05</span>]<span class="sc">$</span>bait.length,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">alternative=</span><span class="st">&quot;greater&quot;</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>non_b2b_frag_wt <span class="ot">&lt;-</span> </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wilcox.test</span>(merged_runs_non_b2b[q.value<span class="sc">&lt;</span><span class="fl">0.05</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>              merged_runs_non_b2b[q.value<span class="sc">&gt;=</span><span class="fl">0.05</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">alternative=</span><span class="st">&quot;greater&quot;</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="do">##bait to bait</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>b2b_bait_wt <span class="ot">&lt;-</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wilcox.test</span>(merged_runs_b2b[q.value<span class="sc">&lt;</span><span class="fl">0.05</span>]<span class="sc">$</span>bait.length,</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>              merged_runs_b2b[q.value<span class="sc">&gt;=</span><span class="fl">0.05</span>]<span class="sc">$</span>bait.length,</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">alternative=</span><span class="st">&quot;greater&quot;</span>)</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>b2b_frag_wt <span class="ot">&lt;-</span> </span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wilcox.test</span>(merged_runs_b2b[q.value<span class="sc">&lt;</span><span class="fl">0.05</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>              merged_runs_b2b[q.value<span class="sc">&gt;=</span><span class="fl">0.05</span>]<span class="sc">$</span>frag.length,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">alternative=</span><span class="st">&quot;greater&quot;</span>)</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(non_b2b_bait_wt)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[q.value &lt; 0.05]$bait.length and merged_runs_non_b2b[q.value &gt;= 0.05]$bait.length</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W = 1.464e+12, p-value = 1</span></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true location shift is greater than 0</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(non_b2b_frag_wt)</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_non_b2b[q.value &lt; 0.05]$frag.length and merged_runs_non_b2b[q.value &gt;= 0.05]$frag.length</span></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W = 1.7498e+12, p-value = 0.9505</span></span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true location shift is greater than 0</span></span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(b2b_bait_wt)</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_b2b[q.value &lt; 0.05]$bait.length and merged_runs_b2b[q.value &gt;= 0.05]$bait.length</span></span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W = 9051619314, p-value = 1</span></span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true location shift is greater than 0</span></span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(b2b_frag_wt)</span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  merged_runs_b2b[q.value &lt; 0.05]$frag.length and merged_runs_b2b[q.value &gt;= 0.05]$frag.length</span></span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W = 9.399e+09, p-value = 1</span></span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true location shift is greater than 0</span></span></code></pre></div>
<p>Show the plot and results</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>plt_non_b2b_frag <span class="ot">&lt;-</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(merged_runs_non_b2b,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">aes</span>(<span class="at">x=</span>signif,<span class="at">y=</span><span class="fu">log</span>(frag.length,<span class="at">base =</span> <span class="dv">10</span>),<span class="at">fill=</span>signif))<span class="sc">+</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>,<span class="at">outlier.shape=</span><span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.1</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T,<span class="at">alpha =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Log 10 length&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Significant Interaction&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Fragment length Non-bait-to-bait interactions&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))<span class="sc">+</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">quantile</span>(<span class="fu">log</span>(merged_runs_non_b2b<span class="sc">$</span>frag.length,<span class="at">base =</span> <span class="dv">10</span>),</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.9</span>)))</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in geom_violin(trim = FALSE, outlier.shape = NA): Ignoring unknown</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; parameters: `outlier.shape`</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>plt_non_b2b_bait <span class="ot">&lt;-</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(merged_runs_non_b2b,</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>         <span class="fu">aes</span>(<span class="at">x=</span>signif,<span class="at">y=</span><span class="fu">log</span>(bait.length,<span class="at">base=</span><span class="dv">10</span>),<span class="at">fill=</span>signif))<span class="sc">+</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>,<span class="at">outlier.shape=</span><span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.1</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T,<span class="at">alpha =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Log 10 Length&quot;</span>) <span class="sc">+</span> </span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Significant Interaction&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Bait length non-bait-to-bait interactions&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in geom_violin(trim = FALSE, outlier.shape = NA): Ignoring unknown</span></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; parameters: `outlier.shape`</span></span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>plt_b2b_frag <span class="ot">&lt;-</span> </span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(merged_runs_b2b,</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>         <span class="fu">aes</span>(<span class="at">x=</span>signif,<span class="at">y=</span><span class="fu">log</span>(frag.length,<span class="at">base =</span> <span class="dv">10</span>),<span class="at">fill=</span>signif))<span class="sc">+</span></span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>,<span class="at">outlier.shape=</span><span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.1</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T,<span class="at">alpha=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Log 10 Length&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Significant Interaction&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Fragment Length Bait-to-bait interactions&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>))</span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in geom_violin(trim = FALSE, outlier.shape = NA): Ignoring unknown</span></span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; parameters: `outlier.shape`</span></span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a>plt_b2b_bait <span class="ot">&lt;-</span> </span>
<span id="cb21-49"><a href="#cb21-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(merged_runs_b2b,</span>
<span id="cb21-50"><a href="#cb21-50" aria-hidden="true" tabindex="-1"></a>         <span class="fu">aes</span>(<span class="at">x=</span>signif,<span class="at">y=</span><span class="fu">log</span>(bait.length,<span class="at">base =</span> <span class="dv">10</span>),<span class="at">fill=</span>signif))<span class="sc">+</span></span>
<span id="cb21-51"><a href="#cb21-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>,<span class="at">outlier.shape=</span><span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-52"><a href="#cb21-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.1</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb21-53"><a href="#cb21-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb21-54"><a href="#cb21-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T,<span class="at">alpha =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb21-55"><a href="#cb21-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-56"><a href="#cb21-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Log 10 Length&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-57"><a href="#cb21-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Significant Interactions&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-58"><a href="#cb21-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Bait length bait-to-bait interactions&quot;</span>)<span class="sc">+</span></span>
<span id="cb21-59"><a href="#cb21-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span>
<span id="cb21-60"><a href="#cb21-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in geom_violin(trim = FALSE, outlier.shape = NA): Ignoring unknown</span></span>
<span id="cb21-61"><a href="#cb21-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; parameters: `outlier.shape`</span></span>
<span id="cb21-62"><a href="#cb21-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-63"><a href="#cb21-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-64"><a href="#cb21-64" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(plt_non_b2b_frag,plt_non_b2b_bait,</span>
<span id="cb21-65"><a href="#cb21-65" aria-hidden="true" tabindex="-1"></a>                          plt_b2b_frag,plt_b2b_bait,<span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-8-1.png" width="672.75552" /></p>
</div>
<div id="distance-of-interactions-from-transcriptional-start-site"
class="section level2">
<h2>Distance of interactions from Transcriptional start site</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(merged_stretch_cis, <span class="fu">aes</span>(<span class="fu">log</span>(distance,<span class="at">base=</span><span class="dv">10</span>))) <span class="sc">+</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>)<span class="sc">+</span><span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">name=</span><span class="st">&quot;Adj. P-value Bin&quot;</span>,<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_viridis</span>(<span class="at">name=</span><span class="st">&quot;Adj. P-value Bin&quot;</span>,<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="fu">log</span>(<span class="dv">24000000</span>,<span class="at">base=</span><span class="dv">10</span>), <span class="at">colour=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="fu">log</span>(<span class="dv">500000</span>,<span class="at">base=</span><span class="dv">10</span>), <span class="at">colour=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">x=</span><span class="fu">log</span>(<span class="dv">24000000</span>,<span class="at">base=</span><span class="dv">10</span>),<span class="at">y=</span><span class="dv">2</span>,<span class="at">label=</span><span class="st">&quot;24Mb&quot;</span>,<span class="at">vjust=</span><span class="dv">2</span>,<span class="at">geom=</span><span class="st">&quot;label&quot;</span>)<span class="sc">+</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">x=</span><span class="fu">log</span>(<span class="dv">500000</span>,<span class="at">base=</span><span class="dv">10</span>),<span class="at">y=</span><span class="dv">2</span>,<span class="at">label=</span><span class="st">&quot;500Kb&quot;</span>,<span class="at">vjust=</span><span class="dv">2</span>,<span class="at">geom=</span><span class="st">&quot;label&quot;</span>)<span class="sc">+</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Log 10 Distance from Interaction to RNA&quot;</span>)<span class="sc">+</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-9-1.png" width="672.75552" /></p>
<p>Split the plot according to the Replicate type and next according to
the p value</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(merged_stretch_cis,<span class="fu">aes</span>(<span class="fu">log</span>(distance,<span class="at">base =</span> <span class="dv">10</span>),</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">fill=</span>q_bin,<span class="at">colour=</span>q_bin))<span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span> <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type)<span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">name=</span><span class="st">&quot;Adj P-value bin&quot;</span>, <span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_viridis</span>(<span class="at">name=</span><span class="st">&quot;Adj P-value bin&quot;</span>, <span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">log</span>(<span class="dv">24000000</span>,<span class="at">base =</span> <span class="dv">10</span>), <span class="at">colour=</span><span class="st">&quot;grey&quot;</span>)<span class="sc">+</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">log</span>(<span class="dv">500000</span>,<span class="at">base =</span> <span class="dv">10</span>), <span class="at">colour=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">x=</span><span class="fu">log</span>(<span class="dv">24000000</span>,<span class="at">base =</span> <span class="dv">10</span>), <span class="at">y=</span><span class="dv">2</span>, <span class="at">label=</span><span class="st">&quot;24Mb&quot;</span>,<span class="at">vjust=</span><span class="dv">2</span>,<span class="at">geom =</span> <span class="st">&quot;label&quot;</span>) <span class="sc">+</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">x=</span><span class="fu">log</span>(<span class="dv">500000</span>,<span class="at">base =</span> <span class="dv">10</span>), <span class="at">y=</span><span class="dv">2</span>, <span class="at">label=</span><span class="st">&quot;500kb&quot;</span>,<span class="at">vjust=</span><span class="dv">2</span>,<span class="at">geom =</span> <span class="st">&quot;label&quot;</span>) <span class="sc">+</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Log 10 Distance from interaction to RNA&quot;</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-10-1.png" width="672.75552" />
Normalizing the plot for RNA distance</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>merged_stretch_cis[,stnd_dist<span class="sc">:</span><span class="er">=</span>distance<span class="sc">/</span>bait.length]</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(merged_stretch_cis, <span class="fu">aes</span>(<span class="fu">log</span>(stnd_dist,<span class="at">base=</span><span class="dv">10</span>))) <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>)<span class="sc">+</span><span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type)<span class="sc">+</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">name=</span><span class="st">&quot;Adj. P-value Bin&quot;</span>,<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_viridis</span>(<span class="at">name=</span><span class="st">&quot;Adj. P-value Bin&quot;</span>,<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="fu">log</span>(<span class="dv">1</span>,<span class="at">base=</span><span class="dv">10</span>), <span class="at">colour=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="fu">log</span>(<span class="dv">120</span>,<span class="at">base=</span><span class="dv">10</span>), <span class="at">colour=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">x=</span><span class="fu">log</span>(<span class="dv">3</span>,<span class="at">base=</span><span class="dv">10</span>),<span class="at">y=</span>.<span class="dv">8</span>,<span class="at">label=</span><span class="st">&quot;RNA Length&quot;</span>,<span class="at">vjust=</span><span class="dv">2</span>,<span class="at">geom=</span><span class="st">&quot;label&quot;</span>)<span class="sc">+</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">x=</span><span class="fu">log</span>(<span class="dv">400</span>,<span class="at">base=</span><span class="dv">10</span>),<span class="at">y=</span>.<span class="dv">8</span>,<span class="at">label=</span><span class="st">&quot;120x RNA Length&quot;</span>,<span class="at">vjust=</span><span class="dv">2</span>,</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom=</span><span class="st">&quot;label&quot;</span>)<span class="sc">+</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Log 10 Distance from Interaction to RNA of RNA Length&quot;</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-11-1.png" width="672.75552" /></p>
</div>
<div
id="check-the-effect-of-rna-length-normalisation-on-the-distribution-of-counts"
class="section level2">
<h2>Check the effect of RNA Length normalisation on the distribution of
counts</h2>
<p>We compare the distribution of counts after normalising for RNA
lengths, in particular we would like to see a higher effect for
Long-NonCoding RNAs, as they account for a high percentage of total and
significant interactions.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>merged_runs<span class="sc">$</span>target.len <span class="ot">&lt;-</span> merged_runs<span class="sc">$</span>target.end<span class="sc">-</span>merged_runs<span class="sc">$</span>target.start</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>merged_runs<span class="sc">$</span>bait.len <span class="ot">&lt;-</span> merged_runs<span class="sc">$</span>bait.end<span class="sc">-</span>merged_runs<span class="sc">$</span>bait.start</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#plot the relationship between bait length and counts</span></span></code></pre></div>
<p><img src="Images/Chicane_baits.trans_plot.png" /><!-- --></p>
<p><img src="Images/Chicane_targets.trans_plot.png" width="2066" /></p>
<p><img src="Images/Chicane_counts_dist.png" /><!-- --></p>
<p><img src="Images/Chicane_qvalue_plot.png" /><!-- --></p>
</div>
<div id="coding-vs-non-coding-rnas" class="section level2">
<h2>Coding vs Non-coding RNAs</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">#add cis and trans interactions</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>merged_runs[<span class="fu">is.na</span>(distance), interaction<span class="sc">:</span><span class="er">=</span><span class="st">&quot;trans&quot;</span>]</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>merged_runs[<span class="sc">!</span><span class="fu">is.na</span>(distance),interaction<span class="sc">:</span><span class="er">=</span><span class="st">&quot;cis&quot;</span>]</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co">#add in gene type </span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_runs<span class="sc">$</span>bait_name)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">&quot;ENSEMBL_MART_ENSEMBL&quot;</span>, <span class="at">host =</span> <span class="st">&quot;https://asia.ensembl.org&quot;</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useDataset</span>(<span class="st">&quot;hsapiens_gene_ensembl&quot;</span>, mart)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>annotLookup <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> mart,</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hgnc_symbol&quot;</span>,</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;entrezgene_id&quot;</span>,</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;gene_biotype&quot;</span>),</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">filter =</span> <span class="st">&quot;hgnc_symbol&quot;</span>,</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">values =</span> genes,</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">uniqueRows=</span><span class="cn">TRUE</span>,</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">useCache =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>annotLookup <span class="ot">&lt;-</span> <span class="fu">as.data.table</span>(annotLookup)</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="co">#gene_biotype</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(annotLookup,</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>         <span class="fu">c</span>(<span class="st">&quot;bait_name&quot;</span>,<span class="st">&quot;entrezgene_id&quot;</span>,<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;gene_biotype&quot;</span>))</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="fu">setkey</span>(merged_runs,bait_name)</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a><span class="fu">setkey</span>(annotLookup, bait_name)</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>merged_runs[annotLookup, gene_biotype<span class="sc">:</span><span class="er">=</span> i.gene_biotype]</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a><span class="co">#clear memory</span></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(genes,annotLookup,mart,baits)</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>res_biotyp <span class="ot">&lt;-</span> merged_runs[<span class="sc">!</span><span class="fu">is.na</span>(gene_biotype),]</span></code></pre></div>
<p>Protein type</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>res_biotyp[<span class="fu">grepl</span>( <span class="st">&quot;protein_coding&quot;</span>, gene_biotype),gene_biotype_hl<span class="sc">:</span><span class="er">=</span><span class="st">&quot;pcRNA&quot;</span>]</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>res_biotyp[<span class="fu">grepl</span>( <span class="st">&quot;pseudogene&quot;</span>, gene_biotype),gene_biotype_hl<span class="sc">:</span><span class="er">=</span><span class="st">&quot;pseudogene&quot;</span>]</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>res_biotyp[<span class="fu">grepl</span>( <span class="st">&quot;TR_&quot;</span>, gene_biotype),gene_biotype_hl<span class="sc">:</span><span class="er">=</span><span class="st">&quot;TR_gene&quot;</span>]</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>res_biotyp[<span class="fu">grepl</span>( <span class="st">&quot;TEC&quot;</span>, gene_biotype),gene_biotype_hl<span class="sc">:</span><span class="er">=</span>gene_biotype]</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>res_biotyp[gene_biotype <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;miRNA&quot;</span>,<span class="st">&quot;miscRNA&quot;</span>,<span class="st">&quot;piRNA&quot;</span>,<span class="st">&quot;rRNA&quot;</span>,<span class="st">&quot;siRNA&quot;</span>,<span class="st">&quot;snRNA&quot;</span>,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;snoRNA&quot;</span>,<span class="st">&quot;tRNA&quot;</span>,<span class="st">&quot;vaultRNA&quot;</span>,<span class="st">&quot;lncRNA&quot;</span>,<span class="st">&quot;misc_RNA&quot;</span>,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;scaRNA&quot;</span>),</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>           gene_biotype_hl<span class="sc">:</span><span class="er">=</span><span class="st">&quot;ncRNA&quot;</span>]</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>res_biotyp[<span class="fu">grepl</span>( <span class="st">&quot;Mt_&quot;</span>, gene_biotype),gene_biotype_hl<span class="sc">:</span><span class="er">=</span><span class="st">&quot;Mt_RNA&quot;</span>]</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>res_biotyp[gene_biotype<span class="sc">==</span><span class="st">&quot;ribozyme&quot;</span>,gene_biotype_hl<span class="sc">:</span><span class="er">=</span>gene_biotype]</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>res_biotyp[gene_biotype<span class="sc">==</span><span class="st">&quot;scRNA&quot;</span>,gene_biotype_hl<span class="sc">:</span><span class="er">=</span>gene_biotype]</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>res_biotyp[<span class="fu">grepl</span>( <span class="st">&quot;IG_&quot;</span>, gene_biotype),gene_biotype_hl<span class="sc">:</span><span class="er">=</span><span class="st">&quot;IG_gene&quot;</span>]</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="co">#update naming to pcRNA</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>res_biotyp[gene_biotype<span class="sc">==</span><span class="st">&quot;protein_coding&quot;</span>,gene_biotype<span class="sc">:</span><span class="er">=</span><span class="st">&quot;pcRNA&quot;</span>]</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>res_biotyp<span class="sc">$</span>control <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(res_biotyp<span class="sc">$</span>type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Control Coding&quot;</span>,<span class="st">&quot;Control NonCoding&quot;</span>), <span class="st">&quot;Control&quot;</span>,<span class="st">&quot;ICM&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_biotyp,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x=</span>gene_biotype_hl,<span class="at">fill=</span>gene_biotype_hl)) <span class="sc">+</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>()<span class="sc">+</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>control)<span class="sc">+</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Total Number of interactions&quot;</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-19-1.png" width="672.75552" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_biotyp[q.value<span class="sc">&lt;</span><span class="fl">0.05</span>,],</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x=</span>gene_biotype_hl,<span class="at">fill=</span>gene_biotype_hl)) <span class="sc">+</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>()<span class="sc">+</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>control)<span class="sc">+</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Number of significant interactions&quot;</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-20-1.png" width="672.75552" /></p>
<p>Proportion of cis coding/non coding interactions</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>merged_counts <span class="ot">&lt;-</span> res_biotyp[q.value<span class="sc">&lt;</span><span class="fl">0.05</span>, .N, by<span class="ot">=</span>.(interaction,gene_biotype_hl, control)]</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="co">#merged_counts$control &lt;- ifelse(merged_counts$type %in% c(&quot;Control Coding&quot;,&quot;Control NonCoding&quot;), &quot;Control&quot;,&quot;ICM&quot;)</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>merged_counts[,all_N <span class="sc">:</span><span class="er">=</span><span class="fu">sum</span>(N), by<span class="ot">=</span>.(interaction, control)]</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>merged_counts[,prop<span class="sc">:</span><span class="er">=</span>(N<span class="sc">/</span>all_N)]</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(merged_counts[control <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>)],<span class="fu">aes</span>(<span class="at">x=</span>gene_biotype_hl,<span class="at">y=</span>prop,<span class="at">fill=</span>gene_biotype_hl)) <span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>interaction)<span class="sc">+</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Proportion &quot;</span>)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(merged_counts[control <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ICM&quot;</span>)],<span class="fu">aes</span>(<span class="at">x=</span>gene_biotype_hl,<span class="at">y=</span>prop,<span class="at">fill=</span>gene_biotype_hl)) <span class="sc">+</span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>interaction)<span class="sc">+</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Proportion &quot;</span>)</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot1,plot2, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>,<span class="st">&quot;ICM&quot;</span>), <span class="at">nrow =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-21-1.png" width="672.75552" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(merged_counts,<span class="fu">aes</span>(<span class="at">x=</span>gene_biotype_hl,<span class="at">y=</span>prop,<span class="at">fill=</span>gene_biotype_hl)) <span class="sc">+</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(control<span class="sc">~</span>interaction, <span class="at">nrow =</span> <span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span>T)<span class="sc">+</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Proportion of significant cis/trans Interactions&quot;</span>)</span></code></pre></div>
<p><img src="norm_files/figure-html/unnamed-chunk-22-1.png" width="672.75552" height="100%" /></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">#import in the annotation file</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>DNA_annot <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;../../GRCh38-cCREs.bed&quot;</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>CTCF_reg <span class="ot">&lt;-</span> DNA_annot[DNA_annot<span class="sc">$</span>V6 <span class="sc">%like%</span> <span class="st">&quot;CTCF&quot;</span>,]</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>prox_enh_reg <span class="ot">&lt;-</span> DNA_annot[DNA_annot<span class="sc">$</span>V6 <span class="sc">%like%</span> <span class="st">&quot;pELS&quot;</span>,]</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>dist_enh_reg <span class="ot">&lt;-</span> DNA_annot[DNA_annot<span class="sc">$</span>V6 <span class="sc">%like%</span> <span class="st">&quot;dELS&quot;</span>,]</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>promoter_reg <span class="ot">&lt;-</span> DNA_annot[DNA_annot<span class="sc">$</span>V6 <span class="sc">%like%</span> <span class="st">&quot;PLS&quot;</span>,]</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>dnase_h3k4me3_red <span class="ot">&lt;-</span> DNA_annot[DNA_annot<span class="sc">$</span>V6 <span class="sc">%like%</span> <span class="st">&quot;DNase-H3K4me3&quot;</span>,]</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="co">#intersect with the DNA loc file</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bedtools.path =</span> <span class="st">&quot;~/bedtools2/bin/&quot;</span>)</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>DNA_loc <span class="ot">&lt;-</span> merged_runs[,<span class="fu">c</span>(<span class="st">&quot;target.chr&quot;</span>,<span class="st">&quot;target.start&quot;</span>,<span class="st">&quot;target.end&quot;</span>)]</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="co">#write the output of dna loc</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(DNA_loc, <span class="at">file =</span> <span class="st">&quot;~/Radicl_Seq/Norm_merged_runs_dna.tsv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> F, <span class="at">col.names =</span> F, <span class="at">row.names =</span> F)</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a><span class="co">#perform bedtools intersect in terminal</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a><span class="co">#ans &lt;- bedtoolsr::bt.intersect(a=DNA_loc,</span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                               b=CTCF_reg,</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                               wa=T,</span></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                               wb=T)</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ans &lt;- ans[,c(1,2,3,9)]</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ans &lt;- ans[!duplicated(ans),]</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the annotated file</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>ans <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;~/Radicl_Seq/mm10_DNA_annot.tsv&quot;</span>)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co">#add annotation results to RadiclSeq results</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ans) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>,<span class="st">&quot;annotation&quot;</span>)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>target.id <span class="ot">&lt;-</span> <span class="fu">paste0</span>(ans<span class="sc">$</span>chr,<span class="st">&quot;:&quot;</span>,ans<span class="sc">$</span>start,<span class="st">&quot;-&quot;</span>,ans<span class="sc">$</span>end)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>type <span class="ot">&lt;-</span> merged_runs[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, merged_runs<span class="sc">$</span>target.id),<span class="st">&quot;type&quot;</span>]</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>interaction <span class="ot">&lt;-</span> merged_runs[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, merged_runs<span class="sc">$</span>target.id),<span class="st">&quot;interaction&quot;</span>]</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>gene_biotype <span class="ot">&lt;-</span> res_biotyp[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, res_biotyp<span class="sc">$</span>target.id),<span class="st">&quot;gene_biotype_hl&quot;</span>]</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>signif <span class="ot">&lt;-</span> res_biotyp[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, res_biotyp<span class="sc">$</span>target.id),<span class="st">&quot;signif&quot;</span>]</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>control <span class="ot">&lt;-</span> res_biotyp[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, res_biotyp<span class="sc">$</span>target.id),<span class="st">&quot;control&quot;</span>]</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>bait_name <span class="ot">&lt;-</span> res_biotyp[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, res_biotyp<span class="sc">$</span>target.id),<span class="st">&quot;bait_name&quot;</span>]</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>q.value <span class="ot">&lt;-</span> res_biotyp[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, res_biotyp<span class="sc">$</span>target.id),<span class="st">&quot;q.value&quot;</span>]</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>q.value <span class="ot">&lt;-</span> merged_runs[<span class="fu">match</span>(ans<span class="sc">$</span>target.id, merged_runs<span class="sc">$</span>target.id),<span class="st">&quot;q.value&quot;</span>]</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a><span class="co">#copyt this sptep for all the remainign annotation types</span></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="co">#adding columns to identify ctcf and other types of interactions</span></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>ctcf <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ans<span class="sc">$</span>annotation <span class="sc">%like%</span> <span class="st">&quot;CTCF&quot;</span>, <span class="st">&quot;Yes&quot;</span>,<span class="st">&quot;No&quot;</span>)</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>prox_enhc <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ans<span class="sc">$</span>annotation <span class="sc">%like%</span> <span class="st">&quot;pELS&quot;</span>,<span class="st">&quot;Yes&quot;</span>,<span class="st">&quot;No&quot;</span>)</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>dist_enhc <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ans<span class="sc">$</span>annotation <span class="sc">%like%</span> <span class="st">&quot;dELS&quot;</span>,<span class="st">&quot;Yes&quot;</span>,<span class="st">&quot;No&quot;</span>)</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>ans<span class="sc">$</span>promoter <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ans<span class="sc">$</span>annotation <span class="sc">%like%</span> <span class="st">&quot;PLS&quot;</span>,<span class="st">&quot;Yes&quot;</span>,<span class="st">&quot;No&quot;</span>)</span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a><span class="co">#making a counts table</span></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>dna_counts[[<span class="st">&quot;CTCF&quot;</span>]] <span class="ot">&lt;-</span> ans[ctcf<span class="sc">==</span><span class="st">&quot;Yes&quot;</span> ,.N, by<span class="ot">=</span>.(interaction, gene_biotype, signif)]</span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>dna_counts[[<span class="st">&quot;prox_enhc&quot;</span>]] <span class="ot">&lt;-</span> ans[prox_enhc<span class="sc">==</span><span class="st">&quot;Yes&quot;</span> ,.N, by<span class="ot">=</span>.(interaction, gene_biotype, signif)]</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>dna_counts[[<span class="st">&quot;dist_enhc&quot;</span>]] <span class="ot">&lt;-</span> ans[dist_enhc<span class="sc">==</span><span class="st">&quot;Yes&quot;</span> ,.N, by<span class="ot">=</span>.(interaction, gene_biotype,signif)]</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>dna_counts[[<span class="st">&quot;promoter&quot;</span>]] <span class="ot">&lt;-</span> ans[promoter<span class="sc">==</span><span class="st">&quot;Yes&quot;</span>,.N, by<span class="ot">=</span>.(interaction, gene_biotype, signif)]</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>annot_counts <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dna_counts<span class="sc">$</span>CTCF, dna_counts<span class="sc">$</span>prox_enhc,dna_counts<span class="sc">$</span>dist_enhc,dna_counts<span class="sc">$</span>promoter)</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>annot_counts <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(annot_counts)</span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a><span class="co">#adding type of dna annot to the counts table</span></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>annot_counts<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;CTCF&quot;</span>,<span class="dv">18</span>), <span class="fu">rep</span>(<span class="st">&quot;prox_enhc&quot;</span>,<span class="dv">16</span>), <span class="fu">rep</span>(<span class="st">&quot;dist_enhc&quot;</span>,<span class="dv">19</span>),<span class="fu">rep</span>(<span class="st">&quot;promoter&quot;</span>,<span class="dv">15</span>))</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a><span class="co">#lst &lt;- lapply(All_News, unlist)</span></span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a><span class="co">#data.frame(lapply(lst, `length&lt;-`, max(lengths(lst))))</span></span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a><span class="co">#find the interactions for MALAT1</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#making the plot of distribution of annotations</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>MALAT1 <span class="ot">&lt;-</span> merged_runs[merged_runs<span class="sc">$</span>bait_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;MALAT1&quot;</span>,<span class="st">&quot;TALAM1&quot;</span>),]</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>MALAT1 <span class="ot">&lt;-</span> MALAT1[,<span class="fu">c</span>(<span class="st">&quot;target.id&quot;</span>,<span class="st">&quot;bait.id&quot;</span>,<span class="st">&quot;count&quot;</span>)]</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>GAPDH <span class="ot">&lt;-</span> merged_runs[merged_runs<span class="sc">$</span>bait_name<span class="sc">==</span><span class="st">&quot;GAPDH&quot;</span>,]</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(annot_counts[annot_counts<span class="sc">$</span>signif<span class="sc">==</span><span class="st">&quot;significant&quot;</span>,],<span class="fu">aes</span>(<span class="at">x=</span>type,<span class="at">y=</span>N,<span class="at">fill=</span>type)) <span class="sc">+</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>interaction)<span class="sc">+</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T)<span class="sc">+</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Counts&quot;</span>)<span class="sc">+</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;CTCF&quot;</span>,<span class="st">&quot;Distal Enhc&quot;</span>,<span class="st">&quot;Promoter&quot;</span>,<span class="st">&quot;Proximal Enhc&quot;</span>))</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a><span class="co">#splitting the plot based on coding/nonCoding rna</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>ComplexHeatmap<span class="sc">::</span><span class="fu">Heatmap</span>(<span class="fu">as.matrix</span>(annot_counts[annot_counts<span class="sc">$</span>interaction<span class="sc">==</span><span class="st">&quot;cis&quot;</span>,<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>] <span class="sc">%&gt;%</span> <span class="fu">spread</span>(gene_biotype, N, <span class="at">fill =</span> <span class="dv">0</span>)),</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>                        <span class="at">column_names_side =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cluster_rows =</span> T,</span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cluster_columns =</span> F,</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>                        <span class="at">column_names_rot =</span> <span class="dv">45</span>,</span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>                        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">7</span>),</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>                        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">6</span>),</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>                        <span class="at">row_names_rot =</span> <span class="dv">0</span>,</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot; &quot;</span>,<span class="at">width =</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">unit</span>(<span class="dv">25</span>,<span class="st">&#39;mm&#39;</span>),</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>                        <span class="at">column_title =</span> <span class="st">&quot;T-cell activation genes shRNA Ifn vs Dox&quot;</span>)</span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(annot_counts[annot_counts<span class="sc">$</span>interaction<span class="sc">==</span><span class="st">&quot;cis&quot;</span> <span class="sc">&amp;</span> annot_counts<span class="sc">$</span>signif<span class="sc">==</span><span class="st">&quot;significant&quot;</span>,],<span class="fu">aes</span>(<span class="at">x =</span> gene_biotype,<span class="at">y =</span> type))<span class="sc">+</span></span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill=</span>N))<span class="sc">+</span></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>N))<span class="sc">+</span></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>()</span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(annot_counts[annot_counts<span class="sc">$</span>signif<span class="sc">==</span><span class="st">&quot;significant&quot;</span>,],<span class="fu">aes</span>(<span class="at">x =</span> gene_biotype,<span class="at">y =</span> type))<span class="sc">+</span></span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill=</span>N))<span class="sc">+</span></span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>N))<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>interaction)<span class="sc">+</span></span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_viridis</span>()</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gtools)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>om <span class="ot">=</span> <span class="fu">circos.par</span>(<span class="st">&quot;track.margin&quot;</span>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>oc <span class="ot">=</span> <span class="fu">circos.par</span>(<span class="st">&quot;cell.padding&quot;</span>)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">track.margin =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">cell.padding =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">start.degree =</span> <span class="sc">-</span><span class="dv">250</span>)</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.initializeWithIdeogram</span>(<span class="at">track.height =</span> <span class="fl">0.05</span>)</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="do">### Labels for inversely changing DMRs with DEG</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39; circos.genomicLabels(Gene_Labels, labels.column=4, side=&#39;outside&#39;, cex=0.38)</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Add CpG Island Lines here (blocks to tell the reader where the CpG Islands are located?)</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Methylation Density</span></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="ot">&lt;-</span><span class="fu">select</span>(GAPDH, <span class="at">chrom=</span>target.chr,</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">chromStart=</span>target.start, <span class="at">perc.change=</span>count, <span class="at">chromEnd=</span>target.end)</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39; DMR.PerChange&lt;-mutate(DMR.PerChange, chromEnd=chromStart+1)</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="ot">&lt;-</span><span class="fu">select</span>(DMR.PerChange, chrom, chromStart, chromEnd, perc.change)</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="sc">$</span>chrom<span class="ot">&lt;-</span><span class="fu">factor</span>(DMR.PerChange<span class="sc">$</span>chrom, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>,</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr5&quot;</span>, <span class="st">&quot;chr6&quot;</span>, <span class="st">&quot;chr7&quot;</span>, <span class="st">&quot;chr8&quot;</span>,</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr9&quot;</span>, <span class="st">&quot;chr10&quot;</span>, <span class="st">&quot;chr11&quot;</span>, <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr13&quot;</span>, <span class="st">&quot;chr14&quot;</span>, <span class="st">&quot;chr15&quot;</span>, <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr17&quot;</span>, <span class="st">&quot;chr18&quot;</span>, <span class="st">&quot;chr19&quot;</span>, <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>, <span class="st">&quot;chr23&quot;</span>, <span class="st">&quot;chrX&quot;</span>,</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chrY&quot;</span>))</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="ot">&lt;-</span>DMR.PerChange[<span class="fu">order</span>(DMR.PerChange<span class="sc">$</span>chrom),]</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Methyl.UP&lt;-filter(DMR.PerChange, perc.change&gt;0)</span></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Methyl.DOWN&lt;-filter(DMR.PerChange, perc.change&lt;0)</span></span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Methyl.List&lt;-list(Methyl.UP, Methyl.DOWN)</span></span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.genomicDensity</span>(GAPDH, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;#FF000080&quot;</span>, <span class="st">&quot;darkgreen&quot;</span>),</span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">track.height=</span><span class="fl">0.1</span>, <span class="at">bg.border=</span><span class="cn">NA</span>)</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a><span class="do">##DEG with inverse GPI Islands Promoters</span></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a><span class="co">#circos.genomicTrackPlotRegion(Gene_FoldChange_List,</span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">track.margin=</span>om, <span class="at">cell.padding=</span>oc)</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Add link for all DEGs with DMRs in promoter CGIs</span></span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>Link_Anchor <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../1_Input/4_Combined/Circos/Link_Anchor.csv&quot;</span>)</span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>Link<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;../1_Input/4_Combined/Circos/Link_DEG.DMR_Promoter.CGI_P&lt;0.05.csv&quot;</span>)</span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>Link<span class="sc">$</span>chrom<span class="ot">&lt;-</span><span class="fu">factor</span>(Link<span class="sc">$</span>chrom, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>,</span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr5&quot;</span>, <span class="st">&quot;chr6&quot;</span>, <span class="st">&quot;chr7&quot;</span>, <span class="st">&quot;chr8&quot;</span>,</span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr9&quot;</span>, <span class="st">&quot;chr10&quot;</span>, <span class="st">&quot;chr11&quot;</span>, <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr13&quot;</span>, <span class="st">&quot;chr14&quot;</span>, <span class="st">&quot;chr15&quot;</span>, <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr17&quot;</span>, <span class="st">&quot;chr18&quot;</span>, <span class="st">&quot;chr19&quot;</span>, <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>, <span class="st">&quot;chr23&quot;</span>, <span class="st">&quot;chrX&quot;</span>,</span>
<span id="cb35-43"><a href="#cb35-43" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chrY&quot;</span>))</span>
<span id="cb35-44"><a href="#cb35-44" aria-hidden="true" tabindex="-1"></a>Link<span class="ot">&lt;-</span>Link[<span class="fu">order</span>(Link<span class="sc">$</span>chrom),]</span>
<span id="cb35-45"><a href="#cb35-45" aria-hidden="true" tabindex="-1"></a>Link_Anchor<span class="ot">&lt;-</span>Link_Anchor[<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(Link),]</span>
<span id="cb35-46"><a href="#cb35-46" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.genomicLink</span>(Link, Link_Anchor, <span class="at">col=</span><span class="st">&quot;black&quot;</span>, <span class="at">lwd=</span><span class="fl">0.5</span>)</span>
<span id="cb35-47"><a href="#cb35-47" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.clear</span>()</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>interaction_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">chr1 =</span> <span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>),</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">start1 =</span> <span class="fu">c</span>(<span class="dv">10000</span>, <span class="dv">20000</span>, <span class="dv">30000</span>, <span class="dv">40000</span>, <span class="dv">50000</span>, <span class="dv">60000</span>),</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">end1 =</span> <span class="fu">c</span>(<span class="dv">15000</span>, <span class="dv">25000</span>, <span class="dv">35000</span>, <span class="dv">45000</span>, <span class="dv">55000</span>, <span class="dv">65000</span>),</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chr2 =</span> <span class="fu">c</span>(<span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>, <span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr1&quot;</span>),</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">start2 =</span> <span class="fu">c</span>(<span class="dv">30000</span>, <span class="dv">40000</span>, <span class="dv">50000</span>, <span class="dv">60000</span>, <span class="dv">10000</span>, <span class="dv">20000</span>),</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">end2 =</span> <span class="fu">c</span>(<span class="dv">35000</span>, <span class="dv">45000</span>, <span class="dv">55000</span>, <span class="dv">65000</span>, <span class="dv">15000</span>, <span class="dv">25000</span>)</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate sample interaction data</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace with your actual interaction data</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>interaction_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">set1 =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;D&quot;</span>),</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">set2 =</span> <span class="fu">c</span>(<span class="st">&quot;B&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;D&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;A&quot;</span>),</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">count =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">1</span>)</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a circos plot using chordDiagram</span></span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>chord_data <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(interaction_data, </span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>                           <span class="at">grid.col =</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(<span class="fu">unique</span>(interaction_data<span class="sc">$</span>set1))))</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Customize the circos plot</span></span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.clear</span>()</span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">track.height =</span> <span class="fl">0.05</span>)</span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.initializeWithIdeogram</span>()</span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the order and colors of the sets</span></span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>sets <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(interaction_data<span class="sc">$</span>set1, interaction_data<span class="sc">$</span>set2))</span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>set_order <span class="ot">&lt;-</span> <span class="fu">match</span>(sets, interaction_data<span class="sc">$</span>set1)</span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>set_colors <span class="ot">&lt;-</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(sets))</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the links to the circos plot</span></span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(chord_data)) {</span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.link</span>(</span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a>    chord_data<span class="sc">$</span>rn[i], chord_data<span class="sc">$</span>o1[i], chord_data<span class="sc">$</span>cn[i], chord_data<span class="sc">$</span>o2[i],</span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> set_colors[chord_data<span class="sc">$</span>rn[i]], <span class="at">border =</span> set_colors[chord_data<span class="sc">$</span>cn[i]],</span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">lwd =</span> <span class="dv">2</span></span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ideograms for the sets</span></span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.genomicIdeogram</span>(</span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">cytoband =</span> <span class="cn">NULL</span>,</span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="cn">NULL</span>,</span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a>  <span class="at">track.height =</span> <span class="fl">0.05</span>,</span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">track.margin =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.05</span>)</span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>om <span class="ot">=</span> <span class="fu">circos.par</span>(<span class="st">&quot;track.margin&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>oc <span class="ot">=</span> <span class="fu">circos.par</span>(<span class="st">&quot;cell.padding&quot;</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">track.margin =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">cell.padding =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">start.degree =</span> <span class="sc">-</span><span class="dv">250</span>)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.initializeWithIdeogram</span>()</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="do">### Labels for inversely changing DMRs with DEG</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co">#circos.genomicLabels(Gene_Labels, labels.column=4, side=&#39;outside&#39;, cex=0.38)</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Add CpG Island Lines here (blocks to tell the reader where the CpG Islands are located?)</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Methylation Density</span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="ot">&lt;-</span>MALAT1[, <span class="fu">c</span>(<span class="st">&quot;target.chr&quot;</span>,<span class="st">&quot;target.start&quot;</span>,<span class="st">&quot;target.end&quot;</span>,<span class="st">&quot;count&quot;</span>)]</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="sc">$</span>chrom<span class="ot">&lt;-</span><span class="fu">factor</span>(DMR.PerChange<span class="sc">$</span>target.chr, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>,<span class="st">&quot;chr5&quot;</span>,</span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>                                                          <span class="st">&quot;chr6&quot;</span>, <span class="st">&quot;chr7&quot;</span>, <span class="st">&quot;chr8&quot;</span>,<span class="st">&quot;chr9&quot;</span>, <span class="st">&quot;chr10&quot;</span>,</span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>                                                          <span class="st">&quot;chr11&quot;</span>,<span class="st">&quot;chr12&quot;</span>,<span class="st">&quot;chr13&quot;</span>, <span class="st">&quot;chr14&quot;</span>,<span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>                                                          <span class="st">&quot;chr16&quot;</span>,<span class="st">&quot;chr17&quot;</span>, <span class="st">&quot;chr18&quot;</span>, <span class="st">&quot;chr19&quot;</span>,<span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>                                                          <span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>, <span class="st">&quot;chr23&quot;</span>, <span class="st">&quot;chrX&quot;</span>,<span class="st">&quot;chrY&quot;</span>))</span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>DMR.PerChange<span class="ot">&lt;-</span>DMR.PerChange[<span class="fu">order</span>(DMR.PerChange<span class="sc">$</span>chrom),]</span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Methyl.UP&lt;-filter(DMR.PerChange, perc.change&gt;0)</span></span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Methyl.DOWN&lt;-filter(DMR.PerChange, perc.change&lt;0)</span></span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Methyl.List&lt;-list(Methyl.UP, Methyl.DOWN)</span></span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39; circos.genomicDensity(DMR.PerChange, col=c(&quot;#FF000080&quot;, &quot;darkgreen&quot;),track.height=0.1, bg.border=NA)</span></span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a><span class="do">##DEG with inverse GPI Islands Promoters</span></span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.genomicTrackPlotRegion</span>(DMR.PerChange[,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>],<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="at">bg.border=</span><span class="cn">NA</span>,</span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>                              <span class="at">panel.fun =</span> <span class="cf">function</span>(region, value, ...) {col <span class="ot">=</span> <span class="fu">ifelse</span>(value[[<span class="dv">1</span>]] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;darkgoldenrod1&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">circos.genomicPoints</span>(region, value, <span class="at">col =</span> col, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>                              cell.xlim <span class="ot">=</span> <span class="fu">get.cell.meta.data</span>(<span class="st">&quot;cell.xlim&quot;</span>)</span>
<span id="cb37-28"><a href="#cb37-28" aria-hidden="true" tabindex="-1"></a>                              <span class="cf">for</span>(h <span class="cf">in</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="sc">-</span><span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">4</span>)) {</span>
<span id="cb37-29"><a href="#cb37-29" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">circos.lines</span>(cell.xlim, <span class="fu">c</span>(h, h), <span class="at">col =</span><span class="st">&quot;#00000040&quot;</span>)</span>
<span id="cb37-30"><a href="#cb37-30" aria-hidden="true" tabindex="-1"></a>                                }</span>
<span id="cb37-31"><a href="#cb37-31" aria-hidden="true" tabindex="-1"></a>                              }, <span class="at">track.height =</span> <span class="fl">0.1</span>)</span>
<span id="cb37-32"><a href="#cb37-32" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">track.margin=</span>om, <span class="at">cell.padding=</span>oc)</span>
<span id="cb37-33"><a href="#cb37-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Add link for all DEGs with DMRs in promoter CGIs</span></span>
<span id="cb37-34"><a href="#cb37-34" aria-hidden="true" tabindex="-1"></a>Link_Anchor <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../1_Input/4_Combined/Circos/Link_Anchor.csv&quot;</span>)</span>
<span id="cb37-35"><a href="#cb37-35" aria-hidden="true" tabindex="-1"></a>Link<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;../1_Input/4_Combined/Circos/Link_DEG.DMR_Promoter.CGI_P&lt;0.05.csv&quot;</span>)</span>
<span id="cb37-36"><a href="#cb37-36" aria-hidden="true" tabindex="-1"></a>Link<span class="sc">$</span>chrom<span class="ot">&lt;-</span><span class="fu">factor</span>(Link<span class="sc">$</span>chrom, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>,</span>
<span id="cb37-37"><a href="#cb37-37" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr5&quot;</span>, <span class="st">&quot;chr6&quot;</span>, <span class="st">&quot;chr7&quot;</span>, <span class="st">&quot;chr8&quot;</span>,</span>
<span id="cb37-38"><a href="#cb37-38" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr9&quot;</span>, <span class="st">&quot;chr10&quot;</span>, <span class="st">&quot;chr11&quot;</span>, <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb37-39"><a href="#cb37-39" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr13&quot;</span>, <span class="st">&quot;chr14&quot;</span>, <span class="st">&quot;chr15&quot;</span>, <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb37-40"><a href="#cb37-40" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr17&quot;</span>, <span class="st">&quot;chr18&quot;</span>, <span class="st">&quot;chr19&quot;</span>, <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb37-41"><a href="#cb37-41" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>, <span class="st">&quot;chr23&quot;</span>, <span class="st">&quot;chrX&quot;</span>,</span>
<span id="cb37-42"><a href="#cb37-42" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;chrY&quot;</span>))</span>
<span id="cb37-43"><a href="#cb37-43" aria-hidden="true" tabindex="-1"></a>Link<span class="ot">&lt;-</span>Link[<span class="fu">order</span>(Link<span class="sc">$</span>chrom),]</span>
<span id="cb37-44"><a href="#cb37-44" aria-hidden="true" tabindex="-1"></a>Link_Anchor<span class="ot">&lt;-</span>Link_Anchor[<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(Link),]</span>
<span id="cb37-45"><a href="#cb37-45" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.genomicLink</span>(MALAT1[,<span class="fu">c</span>(<span class="st">&quot;target.chr&quot;</span>,<span class="st">&quot;target.start&quot;</span>,<span class="st">&quot;target.end&quot;</span>,<span class="st">&quot;count&quot;</span>)],</span>
<span id="cb37-46"><a href="#cb37-46" aria-hidden="true" tabindex="-1"></a>                   MALAT1[,<span class="fu">c</span>(<span class="st">&quot;bait.chr&quot;</span>,<span class="st">&quot;bait.start&quot;</span>,<span class="st">&quot;bait.end&quot;</span>,<span class="st">&quot;count&quot;</span>)], <span class="at">col=</span><span class="st">&quot;black&quot;</span>, <span class="at">lwd=</span><span class="fl">0.5</span>)</span>
<span id="cb37-47"><a href="#cb37-47" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.clear</span>()</span></code></pre></div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
