---
title: "Analysis of RNA length normalized RADICL-Seq data"
author: "Shashank Tiwari"
date: "Most recent update:<br> `r Sys.Date()`"
output: 
  rmarkdown::html_document: 
    theme: spacelab
    highlight: zenburn 
    code_folding: show 
    toc: true 
    toc_float: true
    smooth_scroll: true
    number_sections: false 
    self_contained: true 
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=T, message=F}
#root.dir <- here::here()
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  #root.dir = root.dir
  fig.height = 6,
  fig.width = 7.00787 #178mm
)  
knitr::opts_knit$set(#root.dir = "~/Radicl_Seq/Radicl_seq_functions/", 
                     dpi = 350)  
library(data.table)
library(ggplot2)
library(tidyverse)
library(cowplot)
library(Rsamtools)
library(rtracklayer)
library(viridis)
library(dplyr)
library(genomation)
library(biomaRt)
library(GenomicRanges)
library(DT)
library(magick)
library(patchwork)
library(slickR)
library(svglite)
library(scales)
library(reticulate)
#library(bedtoolsr)
#library(ggpubr)
```
## Running Chicane
Chicane requires three input files, a bam file, a bed file with coordinates of the RNAs and a bed file with binned genome. For our run, I merged the replicates bam file under different conditions and bin the genome into bins of 2kb in length. The RNA coordinates bed file is split into coding and non-coding genes and the overlapping regions are removed.

```{r rna_trim}
knitr::include_graphics("~/Radicl_Seq/RNA_trim.png")

```
Chicane generates a count matrix with the DNA bins and the RNA locations. These counts are then normalized based on the length of the RNA using TPM normalization. Finally a negative binomial model is fit to the counts distribution while accounting for the distance of interactions. This is followed by a multiple testing correction to assign adjusted p-values for each interaction pair



```{r,echo=FALSE, message=FALSE}
load("~/Radicl_Seq/Normalised_2kb_Split/Norm_2kb.RData")
# control_coding <- data.table::fread("../../2kb/control_coding.txt", header = T, sep = ",")
# control_nc <- data.table::fread("../../2kb/control_nc.txt")
# ICM_coding <- data.table::fread("../../2kb/ICM_coding.txt")
# ICM_nc <- data.table::fread("../../2kb/ICM_nc.txt")
#control_coding <- control_coding_2kb
control_nc <- control_nonCoding
#ICM_coding <- ICM_coding_2kb
ICM_nc <- ICM_nonCoding
rm(control_nonCoding, ICM_nonCoding)
gc()

qc_check <- function(significant_results){
  #check propportion by interaction type
  total <- nrow(significant_results)
  trans.prop <- sum(is.na(significant_results$distance))/total
  cis.prop <- sum(!is.na(significant_results$distance))/total
  b2b.prop <- sum(significant_results$bait.to.bait)/total
  int.data <- c('trans' = percent(trans.prop), 'cis' = percent(cis.prop),
                'bait.to.bait' = percent(b2b.prop))
  print(int.data)
}

qc_df <- lapply(list(control_coding, control_nc,ICM_coding,ICM_nc,
                     control_coding[control_coding$q.value<0.05,],control_nc[control_nc$q.value<0.05,],
                     ICM_coding[ICM_coding$q.value<0.05,],
                     ICM_nc[ICM_nc$q.value<0.05,]),
                qc_check)
qc_df <- as.data.frame(qc_df)
Types <- rep(c("Ctrl Coding", "Ctrl NonCoding","ICM Coding","ICM NonCoding"),2)
colnames(qc_df) <- Types


```



```{r}
sktech = htmltools::withTags(table(
  class = 'display',
  thead(
    tr(
      th(rowspan = 2, 'Type'),
      th(colspan = 5, 'All'),
      th(colspan = 5, 'Significant')
    ),
    tr(
      lapply( rep(c("Ctrl Coding", "Ctrl NonCoding","ICM Coding","ICM NonCoding"),2), th)
    )
  )
))

DT::datatable(qc_df, container = sktech, rownames = TRUE,options = list(scrollX=TRUE))


```

We see that trans interactions are the majority across all types. We also see that significant interactions for ICM(both coding and non-coding)  have a slightly higher percentage of trans interactions as compared to others. The `bait-to-bait` interactions are also significantly reduced by normalizing the interactions based on the length of the RNA. 


## Preprocessing 

```{r}
#read in the data
control_coding$type <- as.factor("Control Coding")
control_nc$type <- as.factor("Control NonCoding")
ICM_coding$type <- as.factor("ICM Coding")
ICM_nc$type <- as.factor("ICM NonCoding")
merged_runs <- rbind(control_coding, control_nc,
                     ICM_coding,ICM_nc)
merged_runs$signif <- ifelse(merged_runs$q.value<0.05, TRUE,FALSE)
merged_runs$frag.length <- merged_runs$target.end-merged_runs$target.start
merged_runs$bait.length <- merged_runs$bait.end-merged_runs$bait.start
merged_runs$interaction <- ifelse(is.na(merged_runs$distance),"trans","cis")



#make bins split res by qvalue bins
merged_runs <- merged_runs[!is.na(merged_runs$q.value),]
merged_runs[,q_bin:=cut(q.value, breaks=c(0,0.05,0.2,0.4,0.6,0.8,1),include.lowest=TRUE),]
merged_stretch_cis <- merged_runs[!is.na(distance),.(count2=1:count),names(merged_runs)] 


#get gene names
baits_coding <- fread("~/Radicl_Seq/coding_genes_no_ovrlps.bed")
colnames(baits_coding) <- c("chr","start","end","bait_name")
baits_coding$chr <- sub("^","chr", baits_coding$chr)

baits_noncoding <- fread("~/Radicl_Seq/non_coding_genes_no_ovrlps.bed")
colnames(baits_noncoding) <- c("chr","start","end","bait_name")
baits_noncoding$chr <- sub("^","chr",baits_noncoding$chr)

baits <- rbind(baits_coding, baits_noncoding)
```


```{r,echo=FALSE}
rm(control_coding,control_nc,ICM_coding,ICM_nc)
rm(significant.results.ctrl_coding_2kb,significant.results.ctrl_nc_2kb,significant.results.ICM_coding_2kb,significant.results.ICM_nc_2kb)
gc()
```

## Relationship between RNA length and detected interactions 
To further show the the effect of RNA-length normalization we look at the distribution of adjusted p-value wrt the length of the RNA



```{r, echo=FALSE}
#load the images
knitr::include_graphics("Images/Chicane_baits.trans_plot.png", error = F,dpi = NA )
```

```{r, echo=FALSE}
knitr::include_graphics("Images/Chicane_targets.trans_plot.png", error = FALSE )
```


```{r, echo=FALSE}
knitr::include_graphics("Images/Chicane_counts_dist.png", error = F, dpi = NA)
```

```{r, echo=FALSE}
knitr::include_graphics("Images/Chicane_qvalue_plot.png", error = F, dpi = NA)
```



### Testing the relationship between bait length and p-value

As we have done for our previous runs, we still check for the relationship between significant results and the length of the RNA. Since we also have bait-to-bait interactions, we also check the relatioship between signficant results and the length of the DNA fragments. 
```{r}
merged_runs[,bait_name := baits$bait_name[match(merged_runs$bait.start,baits$start)]]
merged_runs_non_b2b <- merged_runs[bait.to.bait==FALSE]
merged_runs_b2b <- merged_runs[bait.to.bait==TRUE]
setorder(merged_runs_non_b2b,q.value)

#make correlation test
cor.test(merged_runs_non_b2b$bait.length,merged_runs_non_b2b$q.value,
         method="spearman", exact = F)


cor.test(merged_runs_non_b2b$bait.length, merged_runs_non_b2b$count,
         method="spearman", exact = F)


nrow(merged_runs_non_b2b[frag.length==1999])/nrow(merged_runs_non_b2b)

cor.test(merged_runs_non_b2b$frag.length, merged_runs_non_b2b$count,
         method="spearman",exact = F)


cor.test(merged_runs_non_b2b$frag.length, merged_runs_non_b2b$q.value,
         method="spearman",exact=F)


cor.test(merged_runs_non_b2b[frag.length!=1999]$frag.length,
         merged_runs_non_b2b[frag.length!=1999]$count,
         method="spearman",exact=F)


cor.test(merged_runs_non_b2b[frag.length!=1999]$frag.length,
         merged_runs_non_b2b[frag.length!=1999]$q.value,
         method="spearman",exact=F)


#check bait to bait
cor.test(merged_runs_b2b$frag.length, merged_runs_b2b$q.value,
         method="spearman",exact=F)

cor.test(merged_runs_b2b$frag.length, merged_runs_b2b$count,
         method="spearman",exact=F)

#check non-b2b without cis
cor.test(merged_runs_non_b2b[target.chr==bait.chr]$bait.length,
         merged_runs_non_b2b[target.chr==bait.chr]$q.value,
         method="spearman",exact=F)


cor.test(merged_runs_non_b2b[target.chr==bait.chr]$bait.length,
         merged_runs_non_b2b[target.chr==bait.chr]$count,
         method="spearman",exact=F)


#raw pvalue
cor.test(merged_runs_non_b2b[frag.length!=1999]$frag.length,
         merged_runs_non_b2b[frag.length!=1999]$p.value,
         method="spearman",exact=F)


cor.test(merged_runs_non_b2b$frag.length,merged_runs_non_b2b$p.value,
         method="spearman",exact=F)

#summarised counts per fragment
bait_count <- merged_runs_non_b2b[,sum(count),by=.(bait.id,bait.length)]
merged_runs_sum_non_b2b <- 
  cor.test(bait_count$bait.length,bait_count$V1,
           method="spearman",exact=F)
frag_count <- merged_runs_non_b2b[,sum(count),by=.(target.id,frag.length)]
cor.test(frag_count$frag.length,frag_count$V1,
         method="spearman",exact=F)

#bait to bait interactions
bait_count2 <- merged_runs_b2b[,sum(count),by=.(bait.id,bait.length)]
merged_run_sum_frag_b2b <- 
  cor.test(bait_count2$bait.length,bait_count2$V1,
           method="spearman",exact=F)
```

In all of these correlations tests we dont see a relationship between the bait length and significance but we see a higher p-values for the relation between non bait-to-bait DNA fragments and q.value. 


```{r}

#Print summarized counts by bait where it wasnt bait to bait

print(merged_runs_sum_non_b2b)


#Print summ counts by bait where it is bait to bait

print(merged_run_sum_frag_b2b)
```

Next we also try wilcoxons test to compare the distribution of bait lengths and fragment lengths

```{r}
#non bait to bait
non_b2b_bait_wt <-
  wilcox.test(merged_runs_non_b2b[q.value<0.05]$bait.length,
              merged_runs_non_b2b[q.value>=0.05]$bait.length,
              alternative="greater")

non_b2b_frag_wt <- 
  wilcox.test(merged_runs_non_b2b[q.value<0.05]$frag.length,
              merged_runs_non_b2b[q.value>=0.05]$frag.length,
              alternative="greater")

##bait to bait
b2b_bait_wt <-
  wilcox.test(merged_runs_b2b[q.value<0.05]$bait.length,
              merged_runs_b2b[q.value>=0.05]$bait.length,
              alternative="greater")

b2b_frag_wt <- 
  wilcox.test(merged_runs_b2b[q.value<0.05]$frag.length,
              merged_runs_b2b[q.value>=0.05]$frag.length,
              alternative="greater")

print(non_b2b_bait_wt)
print(non_b2b_frag_wt)
print(b2b_bait_wt)
print(b2b_frag_wt)
```

Show the plot and results

```{r}
plt_non_b2b_frag <- 
  ggplot(merged_runs_non_b2b,
         aes(x=signif,y=log(frag.length,base = 10),fill=signif))+
  geom_violin(trim = FALSE,outlier.shape=NA)+
  geom_boxplot(width=0.1, outlier.shape = NA)+
  theme_cowplot()+
  scale_fill_viridis(discrete = T,alpha = 0.5)+
  theme(legend.position="none")+
  ylab("Log 10 length")+
  xlab("Significant Interaction")+
  ggtitle("Fragment length Non-bait-to-bait interactions")+
  theme(plot.title = element_text(size=10))+
  coord_cartesian(ylim = quantile(log(merged_runs_non_b2b$frag.length,base = 10),
                                  c(0.1,0.9)))

plt_non_b2b_bait <-
  ggplot(merged_runs_non_b2b,
         aes(x=signif,y=log(bait.length,base=10),fill=signif))+
  geom_violin(trim = FALSE,outlier.shape=NA)+
  geom_boxplot(width=0.1,outlier.shape = NA)+
  theme_cowplot()+
  scale_fill_viridis(discrete = T,alpha = 0.5)+
  theme(legend.position = "none")+
  ylab("Log 10 Length") + 
  xlab("Significant Interaction")+
  ggtitle("Bait length non-bait-to-bait interactions")+
  theme(plot.title = element_text(size=10))

plt_b2b_frag <- 
  ggplot(merged_runs_b2b,
         aes(x=signif,y=log(frag.length,base = 10),fill=signif))+
  geom_violin(trim = FALSE,outlier.shape=NA)+
  geom_boxplot(width=0.1,outlier.shape = NA)+
  theme_cowplot()+
  scale_fill_viridis(discrete = T,alpha=0.5)+
  theme(legend.position = "none")+
  ylab("Log 10 Length")+
  xlab("Significant Interaction")+
  ggtitle("Fragment Length Bait-to-bait interactions")+
  theme(plot.title = element_text(size = 10))

plt_b2b_bait <- 
  ggplot(merged_runs_b2b,
         aes(x=signif,y=log(bait.length,base = 10),fill=signif))+
  geom_violin(trim = FALSE,outlier.shape=NA)+
  geom_boxplot(width=0.1,outlier.shape = NA)+
  theme_cowplot()+
  scale_fill_viridis(discrete = T,alpha = 0.5)+
  theme(legend.position = "none")+
  ylab("Log 10 Length")+
  xlab("Significant Interactions")+
  ggtitle("Bait length bait-to-bait interactions")+
  theme(plot.title = element_text(size=10))
  

gridExtra::grid.arrange(plt_non_b2b_frag,plt_non_b2b_bait,
                          plt_b2b_frag,plt_b2b_bait,ncol=2)

```






## Distance of interactions from Transcriptional start site

To demonstrate the distance profile for interactions from Chicane, I show a density plot for cis interactions. We also see that Control and ICM Chicane interactions show a different distance distribution profile.

```{r}
ggplot(merged_stretch_cis, aes(log(distance,base=10))) +
    geom_density(alpha = 0.1)+theme_cowplot()+
    scale_fill_viridis(name="Adj. P-value Bin",discrete=T)+
    scale_colour_viridis(name="Adj. P-value Bin",discrete=T)+
    geom_vline(xintercept=log(24000000,base=10), colour="grey") +
    geom_vline(xintercept=log(500000,base=10), colour="grey") +
    annotate(x=log(24000000,base=10),y=2,label="24Mb",vjust=2,geom="label")+
    annotate(x=log(500000,base=10),y=2,label="500Kb",vjust=2,geom="label")+
    xlab("Log 10 Distance from Interaction to RNA")+
  facet_wrap(~type)

```

Split the plot according to the Replicate type and next according to the p value.

```{r}

ggplot(merged_stretch_cis,aes(log(distance,base = 10),
                              fill=q_bin,colour=q_bin))+
  geom_density(alpha=0.1) + theme_cowplot() +
  facet_wrap(~type)+
  scale_fill_viridis(name="Adj P-value bin", discrete=T)+
  scale_colour_viridis(name="Adj P-value bin", discrete=T)+
  geom_vline(xintercept = log(24000000,base = 10), colour="grey")+
  geom_vline(xintercept = log(500000,base = 10), colour="grey") +
  annotate(x=log(24000000,base = 10), y=2, label="24Mb",vjust=2,geom = "label") +
  annotate(x=log(500000,base = 10), y=2, label="500kb",vjust=2,geom = "label") +
  xlab("Log 10 Distance from interaction to RNA")

```

This plot looks to cluttered so I decide to only keep interactions with adjusted p-values<0.05 or p-values>0.8



```{r}

ggplot(merged_stretch_cis[merged_stretch_cis$q.value<0.05 | merged_stretch_cis$q.value>0.8],aes(log(distance,base = 10),
                              fill=q_bin,colour=q_bin))+
  geom_density(alpha=0.1) + theme_cowplot() +
  facet_wrap(~type)+
  scale_fill_viridis(name="Adj P-value bin", discrete=T)+
  scale_colour_viridis(name="Adj P-value bin", discrete=T)+
  geom_vline(xintercept = log(24000000,base = 10), colour="grey")+
  geom_vline(xintercept = log(500000,base = 10), colour="grey") +
  annotate(x=log(24000000,base = 10), y=2, label="24Mb",vjust=2,geom = "label") +
  annotate(x=log(500000,base = 10), y=2, label="500kb",vjust=2,geom = "label") +
  xlab("Log 10 Distance from interaction to RNA")

```

Normalizing the plot for RNA distance



```{r}
merged_stretch_cis[,stnd_dist:=distance/bait.length]


ggplot(merged_stretch_cis, aes(log(stnd_dist,base=10))) +
    geom_density(alpha = 0.1)+theme_cowplot()+
  facet_wrap(~type)+
    scale_fill_viridis(name="Adj. P-value Bin",discrete=T)+
    scale_colour_viridis(name="Adj. P-value Bin",discrete=T)+
    geom_vline(xintercept=log(1,base=10), colour="grey") +
    geom_vline(xintercept=log(120,base=10), colour="grey") +
    annotate(x=log(3,base=10),y=.8,label="RNA Length",vjust=2,geom="label")+
    annotate(x=log(400,base=10),y=.8,label="120x RNA Length",vjust=2,
             geom="label")+
    xlab("Log 10 Distance from Interaction to RNA of RNA Length")

```


```{r}
merged_runs$target.len <- merged_runs$target.end-merged_runs$target.start
merged_runs$bait.len <- merged_runs$bait.end-merged_runs$bait.start
#plot the relationship between bait length and counts
```



## Coding vs Non-coding RNAs

Next we quantify the types of interaction for the RNAs using biomart annotation.



```{r}
#add cis and trans interactions
merged_runs[is.na(distance), interaction:="trans"]
merged_runs[!is.na(distance),interaction:="cis"]
#add in gene type 
genes <- unique(merged_runs$bait_name)

# mart <- useMart("ENSEMBL_MART_ENSEMBL")
# mart <- useDataset("hsapiens_gene_ensembl", mart)
# annotLookup <- getBM(
#   mart = mart,
#   attributes = c(
#     "hgnc_symbol",
#     "entrezgene_id",
#     "ensembl_gene_id",
#     "gene_biotype"),
#   filter = "hgnc_symbol",
#   values = genes,
#   uniqueRows=TRUE,
#   useCache = FALSE)
# 
# annotLookup <- as.data.table(annotLookup)
#saveRDS(annotLookup,file = "annotLookup.RDS")
annotLookup <- readRDS("annotLookup.RDS")

#gene_biotype
setnames(annotLookup,
         c("bait_name","entrezgene_id","ensembl_gene_id","gene_biotype"))

setkey(merged_runs,bait_name)
setkey(annotLookup, bait_name)
merged_runs[annotLookup, gene_biotype:= i.gene_biotype]
#clear memory
rm(genes,annotLookup,mart,baits)
res_biotyp <- merged_runs[!is.na(gene_biotype),]
```

We have 10 million interactions that have a related RNA type in `BiomaRt` which 
can be investigated. These biotype annotations are from [ENSEMBL](http://www.ensembl.org/info/genome/genebuild/biotypes.html). The 
annotations are quite low-level so we are going to roll up as follows:

* Pseudogene: A gene that has homology to known protein-coding genes but contain 
a frameshift and/or stop codon(s) which disrupts the ORF. Thought to have 
arisen through duplication followed by loss of function

* Protein coding: Gene/transcipt that contains an open reading frame (ORF).

* TR gene: T cell receptor gene that undergoes somatic recombination, annotated 
in collaboration with [IMGT](http://www.imgt.org/)

* TEC(To be Experimentally Confirmed): Regions with EST clusters that have polyA
features that could indicate the presence of protein coding genes. These 
require experimental validation, either by 5' RACE or RT-PCR to extend the 
transcripts, or by confirming expression of the putatively-encoded peptide with
specific antibodies.

* ncRNA: A non-coding gene + Long non-coding RNA (lncRNA) A non-coding 
gene/transcript >200bp in length

* Mt_RNA: mitochondrial RNA

* Ribozyme: ribozymal RNA

* scRNA: A small conditional RNA (scRNA) is a small RNA molecule or complex 
(typically less than approximately 100 nt) engineered to interact and change 
conformation conditionally in response to cognate molecular inputs so as to 
perform signal transduction in vitro, in situ, or in vivo.(DEF FROM WIKIPEDIA)

* IG gene: Immunoglobulin gene that undergoes somatic recombination, annotated in
collaboration with IMGT http://www.imgt.org/.




```{r}
res_biotyp[grepl( "protein_coding", gene_biotype),gene_biotype_hl:="pcRNA"]
res_biotyp[grepl( "pseudogene", gene_biotype),gene_biotype_hl:="pseudogene"]
res_biotyp[grepl( "TR_", gene_biotype),gene_biotype_hl:="TR_gene"]
res_biotyp[grepl( "TEC", gene_biotype),gene_biotype_hl:=gene_biotype]
res_biotyp[gene_biotype %in% c("miRNA","miscRNA","piRNA","rRNA","siRNA","snRNA",
                                "snoRNA","tRNA","vaultRNA","lncRNA","misc_RNA",
                                "scaRNA"),
           gene_biotype_hl:="ncRNA"]
res_biotyp[grepl( "Mt_", gene_biotype),gene_biotype_hl:="Mt_RNA"]
res_biotyp[gene_biotype=="ribozyme",gene_biotype_hl:=gene_biotype]
res_biotyp[gene_biotype=="scRNA",gene_biotype_hl:=gene_biotype]
res_biotyp[grepl( "IG_", gene_biotype),gene_biotype_hl:="IG_gene"]
#update naming to pcRNA
res_biotyp[gene_biotype=="protein_coding",gene_biotype:="pcRNA"]
```


### Total interactions

And then we plot the distribution for both the samples while splitting up cis and trans interactions

```{r}
res_biotyp$control <- ifelse(res_biotyp$type %in% c("Control Coding","Control NonCoding"), "Control","ICM")

ggplot(res_biotyp,
        aes(x=gene_biotype_hl,fill=gene_biotype_hl)) +
    geom_bar()+
    theme_cowplot()+
  facet_wrap(~control)+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Total Number of interactions")

```

### Significant interactions

```{r}
ggplot(res_biotyp[q.value<0.05,],
        aes(x=gene_biotype_hl,fill=gene_biotype_hl)) +
    geom_bar()+
    theme_cowplot()+
  facet_wrap(~control)+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Number of significant interactions")
```

### Proportion of cis coding/non coding interactions


```{r}
merged_counts <- res_biotyp[q.value<0.05, .N, by=.(interaction,gene_biotype_hl, control)]
#merged_counts$control <- ifelse(merged_counts$type %in% c("Control Coding","Control NonCoding"), "Control","ICM")
merged_counts[,all_N :=sum(N), by=.(interaction, control)]
merged_counts[,prop:=(N/all_N)]
plot1 <- ggplot(merged_counts[control %in% c("Control")],aes(x=gene_biotype_hl,y=prop,fill=gene_biotype_hl)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Proportion ")

plot2 <- ggplot(merged_counts[control %in% c("ICM")],aes(x=gene_biotype_hl,y=prop,fill=gene_biotype_hl)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Proportion ")
#plot_grid(plot1,plot2, labels = c("Control","ICM"), nrow = 2)
```


```{r, out.height='100%'}
ggplot(merged_counts,aes(x=gene_biotype_hl,y=prop,fill=gene_biotype_hl)) +
    geom_bar(stat="identity")+
    facet_wrap(control~interaction, nrow = 2)+
    theme_cowplot()+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position="none")+
    xlab("")+
    ylab("Proportion of significant cis/trans Interactions")

```



```{r,eval=FALSE,echo=FALSE}
# ICM_coding_norm <- data.table::fread("../../Normalised_2kb_Split/ICM_coding.txt")
# ICM_coding_raw <-data.table::fread("../../Raw_2kb_Split/ICM_coding.txt")
# ICM_merged <- inner_join(ICM_coding_norm,ICM_coding_raw, by=join_by(target.id, bait.id))
# ICM_merged$bait.len <- ICM_merged$bait.end.y-ICM_merged$bait.start.y
# ICM_merged[,bait_name := baits$bait_name[match(ICM_merged$bait.start.x,baits$start)]]
# ICM_merged <- ICM_merged[,c("target.id","bait.id","distance.x","distance.y","count.x","count.y","bait.trans.count.x",
#                             "target.trans.count.x","q.value.x","q.value.y","bait.len", "bait_name")]
# 
# 
# ggplot(ICM_merged, aes(x=log10(bait.len),y=count.x))+
#   geom_density()
# 
# df <- ICM_merged %>% gather(qvalue, var, q.value.x:q.value.y)
# 
# df <- ICM_merged[,c("bait.len","q.value.x", "q.value.y")]%>%
#   pivot_longer(., cols = c(q.value.x, q.value.y),
#                names_to = "Var", values_to = "Val") %>%
#   as.data.table()
# 



ggplot(merged_runs[merged_runs$bait_name=="TALAM1",], 
       aes(x = log(count, base = 10), y = log10(q.value), color= interaction)) + 
    geom_point(size=0.5, alpha=0.5)+
    theme_cowplot()+
    scale_fill_viridis(discrete = T)+
    facet_wrap(~type)

```
## Annotating the interactions for cis-regulatory elements
Using the data from SCREEN:Search Candidate cis-Regulatory Elements by ENCODE, the output from Chicane is overlapped with annotation data. First start of by looking at the overlap between RNA types and annotation status and the plot for all the interactions


```{r,eval=TRUE}
#import in the annotation file

DNA_annot <- data.table::fread("~/Radicl_Seq/GRCh38-cCREs.bed")
#decide which annotation to use for CTCF, either CTCF-bound or CTCF-only
CTCF_reg <- DNA_annot[DNA_annot$V6 %like% "CTCF-only",]
CTCF_reg$type <- "CTCF_reg"
prox_enh_reg <- DNA_annot[DNA_annot$V6 %like% "pELS",]
prox_enh_reg$type <- "Proximal enh_reg"
dist_enh_reg <- DNA_annot[DNA_annot$V6 %like% "dELS",]
dist_enh_reg$type <- "Distal enh_reg"
promoter_reg <- DNA_annot[DNA_annot$V6 %like% "PLS",]
promoter_reg$type <- "Promoter_reg"
dnase_h3k4me3_red <- DNA_annot[DNA_annot$V6 %like% "DNase-H3K4me3",]
dnase_h3k4me3_red$type <- "H3K4Me3_reg"
df <- rbind(CTCF_reg,prox_enh_reg,dist_enh_reg,promoter_reg,dnase_h3k4me3_red)
df <- df[,c(1:3,7)]
names(df) <- c("chr","start","end","annot")
g = GenomicRanges::makeGRangesFromDataFrame(
     df, 
     keep.extra.columns=T, 
     starts.in.df.are.0based=FALSE,
     ignore.strand=is.null(NULL))
annot_grlist <- GenomicRanges::split(g, g$annot, drop = T)
#make granges from chicane output
gres.col <- c("target.chr","target.start","target.end","signif","interaction","gene_biotype_hl","control")
gres = GenomicRanges::makeGRangesFromDataFrame(
    res_biotyp[res_biotyp$type=="Control NonCoding",..gres.col], 
    keep.extra.columns=TRUE, 
    starts.in.df.are.0based=FALSE,
    ignore.strand=TRUE)
gres_list <- GenomicRanges::split(gres,gres$gene_biotype_hl, drop = T)
res_list <- NULL
res_list[["cis"]] <- split(gres_list@unlistData[gres_list@unlistData$interaction=="cis"],gres_list@unlistData$gene_biotype_hl,drop = T)
res_list[["trans"]] <- split(gres_list@unlistData[gres_list@unlistData$interaction=="trans"],gres_list@unlistData$gene_biotype_hl,drop = T)
res_list[["sig"]] <- split(gres_list@unlistData[gres_list@unlistData$signif=="TRUE"],gres_list@unlistData$gene_biotype_hl,drop = T)
res_list[["nonsig"]] <- split(gres_list@unlistData[gres_list@unlistData$signif=="FALSE"],gres_list@unlistData$gene_biotype_hl,drop = T)


rm(gres)
#now get overlap with ChromHMM annotation data - target is radicl-seq data
annotation <- 
    genomation::annotateWithFeatures(gres_list, annot_grlist)
#interested in percentage of target elements overlapping with features:
#make a heatmap
#don't produce heatmap, take matrix, standardise and replot
annot_mtrx <- genomation::heatTargetAnnotation(annotation,plot=FALSE)


#conver to df
annot_df<-reshape2::melt(annot_mtrx)
names(annot_df) <- c("GeneType","State","Overlap")
```

### Gene type vs Annotation state

```{r annotPLot}

chic_dna_plot1 <- ggplot(annot_df,aes(x=State,y=`GeneType`,fill=Overlap))+
  geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))
plot(chic_dna_plot1)
  
```
When looking overall, distal regions are most enriched however the counts are next normalized to annotation type

### Normalize the counts across states

```{r}
#very little notable difference in each state for diff q-values
#try standardising across states
scl_annot_mtrx <- (apply(annot_mtrx,1,rescale,to=c(-1,1)))
#scl_annot_mtrx <-apply(annot_mtrx,2,function(x) scale(x)[,1])
#plot
scl_annot_df<-reshape2::melt(scl_annot_mtrx)
names(scl_annot_df) <- c("GeneType","State","Overlap")
chic_dna_plot2 <- ggplot(scl_annot_df,aes(x=GeneType,y=`State`,fill=Overlap))+
     geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
    scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))
plot(chic_dna_plot2)

#intersect with the DNA loc file
#options(bedtools.path = "~/bedtools2/bin/")
#DNA_loc <- merged_runs[,c("target.chr","target.start","target.end")]
#write the output of dna loc
#fwrite(DNA_loc, file = "~/Radicl_Seq/mm10/Norm_merged_runs_dna.tsv", sep = "\t", quote = F, col.names = F, row.names = F)
#perform bedtools intersect in terminal
#ans <- bedtoolsr::bt.intersect(a=DNA_loc,
#                               b=CTCF_reg,
#                               wa=T,
#                               wb=T)


```

### Normalize the counts across RNA types

```{r}
scl_annot_mtrx <-apply(annot_mtrx,2,function(x) scale(x)[,1])
#plot
scl_annot_df<-reshape2::melt(scl_annot_mtrx)
names(scl_annot_df) <- c("GeneType","State","Overlap")
ggplot(scl_annot_df,aes(x=State,y=`GeneType`,fill=Overlap))+
     geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
    scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))

```

If we only keep the major RNA types we see
```{r}
scl_annot_mtrx <-apply(annot_mtrx[c(1,4,5,7),],2,function(x) scale(x)[,1])
#plot
scl_annot_df<-reshape2::melt(scl_annot_mtrx)
names(scl_annot_df) <- c("GeneType","State","Overlap")
ggplot(scl_annot_df,aes(x=State,y=`GeneType`,fill=Overlap))+
     geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
    scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))

```



### Annotation for cis and trans interaction

```{r}
annotation_extra <- NULL
annotation_extra[["cis"]] <- genomation::annotateWithFeatures(res_list$cis,annot_grlist)
annotation_extra[["trans"]] <- genomation::annotateWithFeatures(res_list$trans,annot_grlist)
annotation_extra[["sig"]] <- genomation::annotateWithFeatures(res_list$sig,annot_grlist)
annotation_extra[["nonsig"]] <- genomation::annotateWithFeatures(res_list$nonsig,annot_grlist)
res_annot_mtrx <- NULL
res_annot_mtrx[["cis"]] <- genomation::heatTargetAnnotation(annotation_extra$cis,plot = F)
res_annot_mtrx[["trans"]] <- genomation::heatTargetAnnotation(annotation_extra$trans,plot=F)
res_annot_mtrx[["sig"]] <- genomation::heatTargetAnnotation(annotation_extra$sig,plot = F)
res_annot_mtrx[["nonsig"]] <- genomation::heatTargetAnnotation(annotation_extra$nonsig,plot = F)

res_annot_df<-reshape2::melt(res_annot_mtrx)
names(res_annot_df) <- c("GeneType","State","Overlap","Type")

ggplot(res_annot_df[res_annot_df$Type %in% c("cis","trans"),],aes(x=State,y=`GeneType`,fill=Overlap))+
   geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  facet_wrap(~Type)+
  ggtitle("All interactions")+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))

ggplot(res_annot_df[res_annot_df$Type=="sig",],aes(x=State,y=`GeneType`,fill=Overlap))+
   geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  ggtitle("Significant interactions")+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))
```




### Annotation for significant interactions

```{r,eval=FALSE}

#res_biotyp[,bait.length:=bait.end-bait.start]
res_biotyp[,stnd_dist:=distance/bait.length]

ggplot(res_biotyp[q.value<0.05 & interaction=="cis" & 
                      (!gene_biotype_hl %in% c("IG_gene","TR_gene"))], 
       aes(x=gene_biotype_hl,y=log(stnd_dist,base=10),fill=gene_biotype_hl)) +
    geom_violin(trim=FALSE)+
    geom_boxplot(width=0.1)+
    theme_cowplot()+
    scale_fill_viridis(discrete=T,alpha=0.5)+
    theme(legend.position="none")+
    ylab("Log 10 Distance from Interaction to RNA")+
    xlab("Interaction Type")

```


```{bash, eval=F,echo=F}
bedtools intersect -a Norm_merged_runs_dna.tsv -b GRCh38-cCREs.bed -wa -wb | cut -f1,2,3,9 | awk '!x[$0]++' | tr ' ' '\t' > /scratch1/users/shashank.tiwari/Radicl_Seq/DNA_annot.tsv


```

### Distribution of total counts for DNA annotations

Plotting the distribution of dna-annotation distribution only for significant interactions, split on the basis of cis and trans interactions

```{r,eval=TRUE,echo=FALSE}
#ans <- ans[,c(1,2,3,9)]
#ans <- ans[!duplicated(ans),]
# read in the annotated file
ans <- fread("~/Radicl_Seq/DNA_annot.bed")
#add annotation results to RadiclSeq results
colnames(ans) <- c("chr","start","end","annotation")
ans$target.id <- paste0(ans$chr,":",ans$start,"-",ans$end)
ans$type <- merged_runs[match(ans$target.id, merged_runs$target.id),"type"]
ans$interaction <- merged_runs[match(ans$target.id, merged_runs$target.id),"interaction"]
ans$gene_biotype <- res_biotyp[match(ans$target.id, res_biotyp$target.id),"gene_biotype_hl"]
ans$signif <- res_biotyp[match(ans$target.id, res_biotyp$target.id),"signif"]
ans$control <- res_biotyp[match(ans$target.id, res_biotyp$target.id),"control"]
ans$bait_name <- res_biotyp[match(ans$target.id, res_biotyp$target.id),"bait_name"]
ans$q.value <- res_biotyp[match(ans$target.id, res_biotyp$target.id),"q.value"]
ans$q.value <- merged_runs[match(ans$target.id, merged_runs$target.id),"q.value"]

#copyt this sptep for all the remainign annotation types
#adding columns to identify ctcf and other types of interactions
ans$ctcf <- ifelse(ans$annotation %like% "CTCF-only", "Yes","No")
ans$prox_enhc <- ifelse(ans$annotation %like% "pELS","Yes","No")
ans$dist_enhc <- ifelse(ans$annotation %like% "dELS","Yes","No")
ans$promoter <- ifelse(ans$annotation %like% "PLS","Yes","No")
ans$H3K4Me3 <- ifelse(ans$annotation %like% "DNase","Yes","No")
dna_counts <- NULL
#making a counts table
dna_counts[["CTCF"]] <- ans[ctcf=="Yes" ,.N, by=.(interaction, gene_biotype, signif)]
dna_counts$CTCF$type <- "CTCF"
dna_counts[["prox_enhc"]] <- ans[prox_enhc=="Yes" ,.N, by=.(interaction, gene_biotype, signif)]
dna_counts$prox_enhc$type <- "prox_enhc"
dna_counts[["dist_enhc"]] <- ans[dist_enhc=="Yes" ,.N, by=.(interaction, gene_biotype,signif)]
dna_counts$dist_enhc$type <- "dist_enhc"
dna_counts[["promoter"]] <- ans[promoter=="Yes",.N, by=.(interaction, gene_biotype, signif)]
dna_counts$promoter$type <-  "promoter"
dna_counts[["H3K3Me3"]] <- ans[H3K4Me3=="Yes",.N, by=.(interaction, gene_biotype, signif)]
dna_counts$H3K3Me3$type <- "H3K4Me3"
annot_counts <- rbind(dna_counts$CTCF, dna_counts$prox_enhc,dna_counts$dist_enhc,dna_counts$promoter,dna_counts$H3K3Me3)
annot_counts <- na.omit(annot_counts)
#adding type of dna annot to the counts table
#annot_counts$type <- c(rep("CTCF",18), rep("prox_enhc",16), rep("dist_enhc",19),rep("promoter",15),rep("H3K4Me3",14))

```




```{r}
ggplot(annot_counts[annot_counts$signif==TRUE],aes(x=type,y=N,fill=type)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete = T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Counts")+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))+
  ggtitle("Significant Interactions")

```
Looking at distribution of only non-significant interactions

```{r nonsig interactions, echo=FALSE}
ggplot(annot_counts[annot_counts$signif==FALSE],aes(x=type,y=N,fill=type)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete = T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Counts")+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))

```


Lastly the interactions for protein coding and noncoding RNAs are plotted.


```{r}
pc <- ggplot(annot_counts[annot_counts$gene_biotype=="pcRNA"],aes(x=type,y=N,fill=type)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete = T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Counts")+
  facet_wrap(~interaction)+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))+
  ggtitle("Protein Coding Genes")

nc <- ggplot(annot_counts[annot_counts$gene_biotype=="ncRNA"],aes(x=type,y=N,fill=type)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete = T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Counts")+
  facet_wrap(~interaction)+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))+
  ggtitle("Non Coding Genes")


pc = xmlSVG({show(pc)}, standalone = TRUE)  
nc = xmlSVG({show(nc)}, standalone = TRUE)  
#p3 = xmlSVG({show(p3)}, standalone = TRUE)  

slickR(list(pc,nc))+settings(dots=TRUE,autoplay = TRUE)

```


## Compare chicane results to binomial model

I ran the binomial model from the original RadiclSeq publication for the ICM samples of the current data set. The Chicane method uses a negative binomial model to model the counts, apart from that Chicane accounts for distace of the interactions and normalizes the counts based on RNA length.

One of the first things to notice was the difference in the numbers of significant interactions from the different models. The binomial model classifies a lot more interactions as significant,however on looking closely we see that instead of the data being binned into 

```{r loadRes}
#load the results from binomial run
icm_binom_all <- data.table::fread("~/RADICL_analysis/2kb/ICM_rna_dna_allInteractions.tsv")
icm_binom_all$signif <- ifelse(icm_binom_all$fdr.tsc<0.05, TRUE,FALSE)
icm_binom_all$frag.length <- icm_binom_all$end.D-icm_binom_all$start.D
icm_binom_all$bait.length <- icm_binom_all$end.R-icm_binom_all$start.R
icm_binom_all$interaction <- ifelse(is.na(icm_binom_all$distance),"trans","cis")


#make bins split res by qvalue bins
icm_binom_all <- icm_binom_all[!is.na(icm_binom_all$fdr.tsc),]
icm_binom_all[,q_bin:=cut(fdr.tsc, breaks=c(0,0.05,0.2,0.4,0.6,0.8,1),include.lowest=TRUE),]
icm_stretch_cis <- icm_binom_all[!is.na(distance),.(count2=1:Count),names(icm_binom_all)] 
```

### Overview of interactions

We can look at the details of total and significant interactions from the binomial model and comapre it with the results from Chicane 
```{r summary}
binom_qc_df <- lapply(list(icm_binom_all,icm_binom_all[fdr.tsc<0.05,]),
                qc_check)
sig_per <- (dim(icm_binom_all[fdr.tsc<0.05,])[1]/dim(icm_binom_all)[1])*100
```

The binomial model identified around 6 percentage interactions from total as significant which is much higher compared to the values from Chicane. Similarly, from the total interactions 68 percent are trans and 32 percent are cis, however when looking at only significant interactions 95 percent of significant interactions are cis interactions. This is completely opposite to the results from Chicane and needs to be investigated further


### Comparing the distribution of distance for significant interactions

Similar to the initial distribution plot, I plot the distribution of distance for cis interactions which are called as significant by the binomial distribution model. I only use the ICM samples to generate these plots

```{r}
plot_binom <- ggplot(icm_stretch_cis, aes(log(distance,base=10))) +
    geom_density(alpha = 0.1)+theme_cowplot()+
    scale_fill_viridis(name="Adj. P-value Bin",discrete=T)+
    scale_colour_viridis(name="Adj. P-value Bin",discrete=T)+
    geom_vline(xintercept=log(24000000,base=10), colour="grey") +
    geom_vline(xintercept=log(500000,base=10), colour="grey") +
    annotate(x=log(24000000,base=10),y=2,label="24Mb",vjust=2,geom="label")+
    annotate(x=log(500000,base=10),y=2,label="500Kb",vjust=2,geom="label")+
    xlab("Log 10 Distance from Interaction to RNA")+
  ggtitle("Binomial Model")

plot_chic <- ggplot(merged_stretch_cis[merged_stretch_cis$type %in% c("ICM NonCoding","ICM Coding"),], aes(log(distance,base=10))) +
    geom_density(alpha = 0.1)+theme_cowplot()+
    scale_fill_viridis(name="Adj. P-value Bin",discrete=T)+
    scale_colour_viridis(name="Adj. P-value Bin",discrete=T)+
    geom_vline(xintercept=log(24000000,base=10), colour="grey") +
    geom_vline(xintercept=log(500000,base=10), colour="grey") +
    annotate(x=log(24000000,base=10),y=2,label="24Mb",vjust=2,geom="label")+
    annotate(x=log(500000,base=10),y=2,label="500Kb",vjust=2,geom="label")+
    xlab("Log 10 Distance from Interaction to RNA")+
  ggtitle("Chicane Model")


binom = xmlSVG({show(plot_binom)}, standalone = TRUE)  
chic = xmlSVG({show(plot_chic)}, standalone = TRUE)  
#p3 = xmlSVG({show(p3)}, standalone = TRUE)  

slickR(list(binom,chic))+settings(dots=TRUE,autoplay = TRUE)
```

Here we see that binomial model upweights the proximal interactions, compared to the Chicane method where along with proximal interactions, distal interactions are also enriched.



Next, we plot the interaction for all interactions split on the p-value

```{r}
#merged_runs[,bait_name := baits$bait_name[match(merged_runs$bait.start,baits$start)]]


plot_binom <- ggplot(icm_stretch_cis,aes(log(distance,base = 10),
                              fill=q_bin,colour=q_bin))+
  geom_density(alpha=0.1) + theme_cowplot() +
  scale_fill_viridis(name="Adj P-value bin", discrete=T)+
  scale_colour_viridis(name="Adj P-value bin", discrete=T)+
  geom_vline(xintercept = log(24000000,base = 10), colour="grey")+
  geom_vline(xintercept = log(500000,base = 10), colour="grey") +
  annotate(x=log(24000000,base = 10), y=2, label="24Mb",vjust=2,geom = "label") +
  annotate(x=log(500000,base = 10), y=2, label="500kb",vjust=2,geom = "label") +
  xlab("Log 10 Distance from interaction to RNA")+
  ggtitle("Binomial Model")

plot_chic <- ggplot(merged_stretch_cis,aes(log(distance,base = 10),
                              fill=q_bin,colour=q_bin))+
  geom_density(alpha=0.1) + theme_cowplot() +
  scale_fill_viridis(name="Adj P-value bin", discrete=T)+
  scale_colour_viridis(name="Adj P-value bin", discrete=T)+
  geom_vline(xintercept = log(24000000,base = 10), colour="grey")+
  geom_vline(xintercept = log(500000,base = 10), colour="grey") +
  annotate(x=log(24000000,base = 10), y=2, label="24Mb",vjust=2,geom = "label") +
  annotate(x=log(500000,base = 10), y=2, label="500kb",vjust=2,geom = "label") +
  xlab("Log 10 Distance from interaction to RNA")+
  ggtitle("Chicane Model")

binom = xmlSVG({show(plot_binom)}, standalone = TRUE)  
chic = xmlSVG({show(plot_chic)}, standalone = TRUE)  
#p3 = xmlSVG({show(p3)}, standalone = TRUE)  

slickR(list(binom,chic))+settings(dots=TRUE,autoplay = TRUE)
```



### RNA length noramlised comparison

Next we normalise the distance based on the length of the RNA


```{r,echo=FALSE}
icm_stretch_cis[,stnd_dist:=distance/bait.length]


plot_binom <- ggplot(icm_stretch_cis, aes(log(stnd_dist,base=10))) +
    geom_density(alpha = 0.1)+theme_cowplot()+
    scale_fill_viridis(name="Adj. P-value Bin",discrete=T)+
    scale_colour_viridis(name="Adj. P-value Bin",discrete=T)+
    geom_vline(xintercept=log(1,base=10), colour="grey") +
    geom_vline(xintercept=log(120,base=10), colour="grey") +
    annotate(x=log(3,base=10),y=.8,label="RNA Length",vjust=2,geom="label")+
    annotate(x=log(400,base=10),y=.8,label="120x RNA Length",vjust=2,
             geom="label")+
    xlab("Log 10 Distance from Interaction to RNA of RNA Length")+
  ggtitle("Binomial Model")


plot_chic <- ggplot(merged_stretch_cis, aes(log(stnd_dist,base=10))) +
    geom_density(alpha = 0.1)+theme_cowplot()+
    scale_fill_viridis(name="Adj. P-value Bin",discrete=T)+
    scale_colour_viridis(name="Adj. P-value Bin",discrete=T)+
    geom_vline(xintercept=log(1,base=10), colour="grey") +
    geom_vline(xintercept=log(120,base=10), colour="grey") +
    annotate(x=log(3,base=10),y=.8,label="RNA Length",vjust=2,geom="label")+
    annotate(x=log(400,base=10),y=.8,label="120x RNA Length",vjust=2,
             geom="label")+
    xlab("Log 10 Distance from Interaction to RNA of RNA Length")+
  ggtitle("Chicane Model")

binom = xmlSVG({show(plot_binom)}, standalone = TRUE)  
chic = xmlSVG({show(plot_chic)}, standalone = TRUE)  
#p3 = xmlSVG({show(p3)}, standalone = TRUE)  

slickR(list(binom,chic))+settings(dots=TRUE,autoplay = TRUE)

```


### Annotation of the RNAs

I annotate the RNAs similar to the Chicane output results, the plot shows the distribution of RNA types for significant interactions.

```{r annotationPlots}
#icm_binom_all$interaction <- ifelse(is.na(icm_binom_all$distance),"trans","cis")
output_counts <- icm_binom_all[fdr.tsc<0.05, .N, by=.(interaction,gene_type.R,annotation.R)]
output_counts[,all_N :=sum(N), by=.(interaction)]
output_counts[,prop:=(N/all_N)]


ggplot(output_counts,aes(x=annotation.R,y=prop,fill=annotation.R) )+
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Proportion ")
```


What we see is that trans interactions are more enriched lncRNAs and cis interations are rich in pcRNAs, however we remove the interactions from lncRNAs, MALAT1 and NEAT1 since they are significantly higher in counts compared to other RNAs.

```{r annotationPlotslnrna}
`%nin%` = Negate(`%in%`)
#assign proper ensembl ids
icm_binom_all$gene_id.R <- sub('\\.[0-9]*$', '', icm_binom_all$gene_id.R)


# mart <- useMart("ENSEMBL_MART_ENSEMBL", host = "https://asia.ensembl.org")
# mart <- useDataset("hsapiens_gene_ensembl", mart)
# gene_IDs <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),
#                   values = icm_binom_all$gene_id.R, mart= mart)
# 
# icm_binom_all$gene_name.R <- gene_IDs[match(icm_binom_all$gene_id.R,gene_IDs[,1]),2]


output_counts <- icm_binom_all[fdr.tsc<0.05 & gene_id.R %nin% c("ENSG00000251562","ENSG00000251562"), 
                               .N, by=.(interaction,gene_type.R,annotation.R)]
output_counts[,all_N :=sum(N), by=.(interaction)]
output_counts[,prop:=(N/all_N)]
ggplot(output_counts,aes(x=annotation.R,y=prop,fill=annotation.R) )+
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete=T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Proportion ")+
  ggtitle("Significant interactions without MALAT1 and NEAT1")
```
So we can see that due the abundance of MALAT1 and NEAT1 interactions in trans, they skew the percentages of lncRNAs.


### Annotating DNA regions for Binomial model

Add DNA-annotations for binomial model and compare the results with Chicane model

```{r}
DNA_annot <- data.table::fread("~/Radicl_Seq/GRCh38-cCREs.bed")
#decide which annotation to use for CTCF, either CTCF-bound or CTCF-only
CTCF_reg <- DNA_annot[DNA_annot$V6 %like% "CTCF-only",]
CTCF_reg$type <- "CTCF_reg"
prox_enh_reg <- DNA_annot[DNA_annot$V6 %like% "pELS",]
prox_enh_reg$type <- "Proximal enh_reg"
dist_enh_reg <- DNA_annot[DNA_annot$V6 %like% "dELS",]
dist_enh_reg$type <- "Distal enh_reg"
promoter_reg <- DNA_annot[DNA_annot$V6 %like% "PLS",]
promoter_reg$type <- "Promoter_reg"
dnase_h3k4me3_red <- DNA_annot[DNA_annot$V6 %like% "DNase-H3K4me3",]
dnase_h3k4me3_red$type <- "H3K4Me3_reg"
df <- rbind(CTCF_reg,prox_enh_reg,dist_enh_reg,promoter_reg,dnase_h3k4me3_red)
df <- df[,c(1:3,7)]
names(df) <- c("chr","start","end","annot")
g = GenomicRanges::makeGRangesFromDataFrame(
     df, 
     keep.extra.columns=T, 
     starts.in.df.are.0based=FALSE,
     ignore.strand=is.null(NULL))
annot_grlist <- GenomicRanges::split(g, g$annot, drop = T)
#make granges from chicane output
gres.col <- c("chr.D","start.D","end.D","signif","interaction","annotation.R")
gres = GenomicRanges::makeGRangesFromDataFrame(
    icm_binom_all[,..gres.col],
    start.field = "start.D",
    end.field = "end.D",
    seqnames.field = "chr.D",
    keep.extra.columns=TRUE, 
    starts.in.df.are.0based=FALSE,
    ignore.strand=TRUE)
gres_list_binom <- GenomicRanges::split(gres,gres$annotation.R, drop = T)
rm(gres)
#now get overlap with ChromHMM annotation data - target is radicl-seq data
annotation_binom <- 
    genomation::annotateWithFeatures(gres_list_binom, annot_grlist)
#interested in percentage of target elements overlapping with features:
#make a heatmap
#don't produce heatmap, take matrix, standardise and replot
annot_mtrx_binom <- genomation::heatTargetAnnotation(annotation_binom,plot=FALSE)


#conver to df
annot_df_binom<-reshape2::melt(annot_mtrx_binom)
names(annot_df_binom) <- c("GeneType","State","Overlap")

```

Overall percentage

```{r}
binom_dna_plot <- ggplot(annot_df_binom,aes(x=State,y=`GeneType`,fill=Overlap))+
  geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))
#plot(binom_dna_plot)
binom = xmlSVG({show(binom_dna_plot)}, standalone = TRUE)  
chic = xmlSVG({show(chic_dna_plot1)}, standalone = TRUE)  
#p3 = xmlSVG({show(p3)}, standalone = TRUE)  

slickR(list(binom,chic))+settings(dots=TRUE,autoplay = TRUE)

```



Next we normalize based on the RNA types

```{r}
scl_annot_mtrx_binom <-apply(annot_mtrx_binom[c(2:5),],2,function(x) scale(x)[,1])
#plot
scl_annot_df_binom<-reshape2::melt(scl_annot_mtrx_binom)
names(scl_annot_df_binom) <- c("GeneType","State","Overlap")
binom_dna_plot_2 <- ggplot(scl_annot_df_binom,aes(x=State,y=`GeneType`,fill=Overlap))+
     geom_tile()+theme_cowplot()+scale_fill_viridis_c()+
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
    scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))
plot(binom_dna_plot_2)
binom = xmlSVG({show(binom_dna_plot_2)}, standalone = TRUE)  
chic = xmlSVG({show(chic_dna_plot2)}, standalone = TRUE)  
#p3 = xmlSVG({show(p3)}, standalone = TRUE)  

slickR(list(binom,chic))+settings(dots=TRUE,autoplay = TRUE)
```




```{r}


#ans <- ans[,c(1,2,3,9)]
#ans <- ans[!duplicated(ans),]
# read in the annotated file
ans <- fread("~/RADICL_analysis/2kb/Binom_Dna_annot.tsv")
#add annotation results to RadiclSeq results
colnames(ans) <- c("chr","start","end","annotation")
icm_binom_all$target.id <- paste0(icm_binom_all$chr.D,":",icm_binom_all$start.D,"-",icm_binom_all$end.D)
ans$target.id <- paste0(ans$chr,":",ans$start,"-",ans$end)

ans$interaction <- icm_binom_all[match(ans$target.id, icm_binom_all$target.id),"interaction"]
ans$gene_biotype <- icm_binom_all[match(ans$target.id, icm_binom_all$target.id),"annotation.R"]
ans$signif <- icm_binom_all[match(ans$target.id, icm_binom_all$target.id),"signif"]
ans$Count <- icm_binom_all[match(ans$target.id, icm_binom_all$target.id),"Count"]
ans$distance <- icm_binom_all[match(ans$target.id, icm_binom_all$target.id),"distance"]

ans$q.value <- icm_binom_all[match(ans$target.id, icm_binom_all$target.id),"fdr.tsc"]

#copyt this sptep for all the remainign annotation types
#adding columns to identify ctcf and other types of interactions
ans$ctcf <- ifelse(ans$annotation %like% "CTCF-only", "Yes","No")
ans$prox_enhc <- ifelse(ans$annotation %like% "pELS","Yes","No")
ans$dist_enhc <- ifelse(ans$annotation %like% "dELS","Yes","No")
ans$promoter <- ifelse(ans$annotation %like% "PLS","Yes","No")
ans$H3K4Me3 <- ifelse(ans$annotation %like% "DNase","Yes","No")
dna_counts <- NULL
#making a counts table
dna_counts[["CTCF"]] <- ans[ctcf=="Yes" ,.N, by=.(interaction, gene_biotype, signif)]
dna_counts$CTCF$type <- "CTCF"
dna_counts[["prox_enhc"]] <- ans[prox_enhc=="Yes" ,.N, by=.(interaction, gene_biotype, signif)]
dna_counts$prox_enhc$type <- "prox_enhc"
dna_counts[["dist_enhc"]] <- ans[dist_enhc=="Yes" ,.N, by=.(interaction, gene_biotype,signif)]
dna_counts$dist_enhc$type <- "dist_enhc"
dna_counts[["promoter"]] <- ans[promoter=="Yes",.N, by=.(interaction, gene_biotype, signif)]
dna_counts$promoter$type <-  "promoter"
dna_counts[["H3K3Me3"]] <- ans[H3K4Me3=="Yes",.N, by=.(interaction, gene_biotype, signif)]
dna_counts$H3K3Me3$type <- "H3K4Me3"
annot_counts <- rbind(dna_counts$CTCF, dna_counts$prox_enhc,dna_counts$dist_enhc,dna_counts$promoter,dna_counts$H3K3Me3)
annot_counts <- na.omit(annot_counts)


# 
# ##dna_counts[["CTCF"]] <- ans[ctcf=="Yes" ,.N, by=.(interaction, gene_biotype, signif)]
# dna_counts[["prox_enhc"]] <- ans[prox_enhc=="Yes" ,.N, by=.(interaction, gene_biotype, signif)]
# dna_counts[["dist_enhc"]] <- ans[dist_enhc=="Yes" ,.N, by=.(interaction, gene_biotype,signif)]
# dna_counts[["promoter"]] <- ans[promoter=="Yes",.N, by=.(interaction, gene_biotype, signif)]
# dna_counts[["H3K3Me3"]] <- ans[H3K4Me3=="Yes",.N, by=.(interaction, gene_biotype, signif)]
# annot_counts <- rbind(dna_counts$CTCF, dna_counts$prox_enhc,dna_counts$dist_enhc,dna_counts$promoter,dna_counts$H3K3Me3)
# annot_counts <- na.omit(annot_counts)
# #adding type of dna annot to the counts table
# annot_counts$type <- c(rep("CTCF",13), rep("prox_enhc",15), rep("dist_enhc",15),rep("promoter",15),rep("H3K4Me3",13))

```




```{r}
ggplot(annot_counts[annot_counts$signif==TRUE],aes(x=type,y=N,fill=type)) +
    geom_bar(stat="identity")+
    facet_wrap(~interaction)+
    theme_cowplot()+
    scale_fill_viridis(discrete = T)+
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
            legend.position="none")+
    xlab("")+
    ylab("Counts")+
  scale_x_discrete(labels=c("CTCF","Distal Enhc","DNase-H3K4me3","Promoter","Proximal Enhc"))+
  ggtitle("Significant Interactions")

```


### Common significant interactions from Chicane and Binomial model


